{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "import os\n",
    "import re\n",
    "from collections import OrderedDict\n",
    "import numpy as np\n",
    "\n",
    "database = '/home/ruidong/Documents/Research/Cancer_Genomics/Data/Databases/Gene_ID_conversion/'\n",
    "os.chdir('/home/ruidong/Documents/Research/share/methylation/')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "fh = open('methylation.beta_value27_and_traits.paired.TCGA-STAD.txt', 'wt')\n",
    "\n",
    "with open('methylation.beta_value27_and_traits.TCGA-STAD.txt') as f:\n",
    "    for line in f:\n",
    "        line = line.rstrip()\n",
    "        lst = line.split('\\t')\n",
    "        if lst[0] == '':\n",
    "            sortedList = sorted(lst)\n",
    "            samples = sortedList[1:sortedList.index('TCGA-BR-4376-11A')+1]\n",
    "            #samples = [sam for sam in samples if sam.endswith('11A')] + [sam for sam in samples if sam.endswith('01A')]\n",
    "            indexSam = [lst.index(sam) for sam in samples]\n",
    "            \n",
    "            #print (indexSam)\n",
    "            #for sam in samples:\n",
    "            #    methyDa[sam] = []\n",
    "                \n",
    "        fh.write(lst[0]+'\\t'+'\\t'.join([lst[i] for i in indexSam])+'\\n')\n",
    "        \n",
    "\n",
    "fh.close()\n",
    "        \n",
    "        "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/ruidong/Tools/anaconda3/lib/python3.5/site-packages/numpy/lib/nanfunctions.py:326: RuntimeWarning: All-NaN slice encountered\n",
      "  warnings.warn(\"All-NaN slice encountered\", RuntimeWarning)\n"
     ]
    }
   ],
   "source": [
    "############# input for survival analysis\n",
    "\n",
    "def therapy_info(x):\n",
    "    x = np.array(x)\n",
    "    y = ['NA']*x.shape[1]\n",
    "    for i in range(x.shape[1]):\n",
    "        z = x[:,i]\n",
    "        if 'YES' in z or 'Chemotherapy' in z:\n",
    "            y[i] = 'YES'\n",
    "        else:\n",
    "            y[i] = 'NO'\n",
    "    return (y)\n",
    "\n",
    "\n",
    "def max_trait(x):\n",
    "    x = np.array(x)\n",
    "    x[x == 'NA'] = np.nan\n",
    "    x = x.astype(float)\n",
    "    x = np.nanmax(x, axis=0)\n",
    "    x = ['NA' if np.isnan(v) else str(v).replace('.0', '') for v in x]\n",
    "    return (x)\n",
    "\n",
    "\n",
    "\n",
    "fh = open('Clinical_data.clean.TCGA-STADxxx.txt', 'wt')\n",
    "\n",
    "survivalDa = OrderedDict()\n",
    "\n",
    "for trt in ['chemo','radiation','targeted_molecular','days_to_last_followup','days_to_death']:\n",
    "    survivalDa[trt]=[]\n",
    "\n",
    "\n",
    "traits = ['gender','ethnicity','race','pathologic_stage', \n",
    "          'pathologic_T', 'pathologic_N', 'pathologic_M']\n",
    "\n",
    "\n",
    "         \n",
    "with open('Merged_clinical_data.TCGA-STAD.txt') as f:\n",
    "    for line in f:\n",
    "        line = line.rstrip()\n",
    "        lst = line.split('\\t')\n",
    "        \n",
    "        if lst[0] == '':\n",
    "            samples = [sam+'-01A' for sam in lst if sam.startswith('TCGA')]\n",
    "            fh.write('\\t'+'\\t'.join(samples)+'\\n')\n",
    "        elif lst[0] == 'age_at_initial_pathologic_diagnosis':\n",
    "            fh.write('age'+'\\t'+'\\t'.join(lst[1:])+'\\n')\n",
    "        elif lst[0] in traits:\n",
    "            lst = [trt.replace(' ', '') for trt in lst]\n",
    "            fh.write('\\t'.join(lst)+'\\n')\n",
    "            \n",
    "        elif lst[0].startswith('days_to_last_followup'):\n",
    "            survivalDa['days_to_last_followup'].append(lst[1:])\n",
    "            \n",
    "        elif lst[0].startswith('days_to_death'):\n",
    "            survivalDa['days_to_death'].append(lst[1:])\n",
    "            \n",
    "            \n",
    "        elif lst[0].startswith('therapy_type'):\n",
    "            survivalDa['chemo'].append(lst[1:])\n",
    "            \n",
    "        elif lst[0].startswith('radiation_therapy'):\n",
    "            survivalDa['radiation'].append(lst[1:])\n",
    "            \n",
    "        elif lst[0].startswith('targeted_molecular_therapy'):\n",
    "            survivalDa['targeted_molecular'].append(lst[1:])        \n",
    "\n",
    "            \n",
    "for ke in ['chemo','radiation','targeted_molecular']:\n",
    "    fh.write(ke+'\\t'+'\\t'.join(therapy_info(survivalDa[ke]))+'\\n')\n",
    "    \n",
    "for ke in ['days_to_death','days_to_last_followup']:\n",
    "    fh.write(ke+'\\t'+'\\t'.join(max_trait(survivalDa[ke]))+'\\n')\n",
    "\n",
    "\n",
    "vital_status = [0 if x == 'NA' else 1 for x in max_trait(survivalDa['days_to_death'])]\n",
    "fh.write('vital_status'+'\\t'+'\\t'.join(map(str,vital_status))+'\\n')\n",
    "\n",
    "\n",
    "fh.close()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 111,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "################################\n",
    "### DM CG SITES\n",
    "\n",
    "sigCG = []\n",
    "with open('methylation27_test.TCGA-STAD.txt') as f:\n",
    "    for line in f:\n",
    "        line = line.rstrip()\n",
    "        lst = line.split('\\t')\n",
    "        if lst[0] == 'Tumor_Beta':\n",
    "            continue\n",
    "        sigCG.append(lst[0])\n",
    "        \n",
    "        "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 117,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "##################################################\n",
    "################## INPUT FOR SURVIVAL ANALYSIS\n",
    "\n",
    "fh = open('methylation.beta_value27_and_traits.survival.TCGA-STAD.txt', 'wt')\n",
    "\n",
    "with open('methylation.beta_value27_and_traits.TCGA-STAD.txt') as f:\n",
    "    for line in f:\n",
    "        line = line.rstrip()\n",
    "        lst = line.split('\\t')\n",
    "        if lst[0] == '':\n",
    "            samples = [sam for sam in lst if re.search('-01A$', sam)]\n",
    "            break\n",
    "            \n",
    "            \n",
    "with open('Clinical_data.clean.TCGA-STAD.txt') as f:\n",
    "    for line in f:\n",
    "        line = line.rstrip()\n",
    "        lst = line.split('\\t')\n",
    "        if lst[0] == '':\n",
    "            samIndex = [lst.index(sam) for sam in samples if sam in lst]\n",
    "            finalSam = [lst[i] for i in samIndex]\n",
    "            \n",
    "        trait = [lst[i] for i in samIndex]\n",
    "        \n",
    "        fh.write(lst[0]+'\\t'+'\\t'.join(trait)+'\\n')\n",
    "        \n",
    "        \n",
    "with open('methylation.beta_value27_and_traits.TCGA-STAD.txt') as f:\n",
    "    for line in f:\n",
    "        line = line.rstrip()\n",
    "        lst = line.split('\\t')\n",
    "        if lst[0] == '':\n",
    "            cgSamIndex = [lst.index(sam) for sam in finalSam]\n",
    "            \n",
    "        elif lst[0] in sigCG:\n",
    "            cgValue = [lst[i] for i in cgSamIndex]\n",
    "            fh.write(lst[0]+'\\t'+'\\t'.join(cgValue)+'\\n')\n",
    "\n",
    "fh.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "####################################################################\n",
    "############## survival inpput (normal)\n",
    "\n",
    "sigCG = []\n",
    "with open('methylation27_test.TCGA-STAD.txt') as f:\n",
    "    for line in f:\n",
    "        line = line.rstrip()\n",
    "        lst = line.split('\\t')\n",
    "        if lst[0] == 'Tumor_Beta':\n",
    "            continue\n",
    "        sigCG.append(lst[0])\n",
    "        \n",
    "         \n",
    "traits = ['sample_type','sample_type_id','tumor_stage','tumor_grade',\n",
    "          'age_at_diagnosis','days_to_death','days_to_last_follow_up',\n",
    "          'vital_status','race','gender','ethnicity']\n",
    "\n",
    "fh = open('methylation.beta_value450_and_traits.survival.TCGA-STAD.txt', 'wt')\n",
    "\n",
    "with open('methylation.beta_value450_and_traits.TCGA-STAD.txt') as f:\n",
    "    for line in f:\n",
    "        line = line.rstrip()\n",
    "        lst = line.split('\\t')\n",
    "        if lst[0] == '':\n",
    "            samples = [sam for sam in lst if re.search('-01A$', sam)]\n",
    "            samIndex = [lst.index(sam) for sam in samples]\n",
    "            fh.write('\\t'+'\\t'.join(samples)+'\\n')\n",
    "        elif lst[0] in traits:\n",
    "            trt = [lst[i] for i in samIndex]\n",
    "            fh.write(lst[0]+'\\t'+'\\t'.join(trt)+'\\n')\n",
    "        elif lst[0] in sigCG:\n",
    "            cgValue = [lst[i] for i in samIndex]\n",
    "            fh.write(lst[0]+'\\t'+'\\t'.join(cgValue)+'\\n')\n",
    "        \n",
    "\n",
    "fh.close()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "###########################################\n",
    "### organize methylation annotation\n",
    "\n",
    "fh = open('methylation450_annotation.clean.txt', 'wt')\n",
    "with open('methylation450_annotation.txt') as f:\n",
    "    for line in f:\n",
    "        line = line.rstrip()\n",
    "        lst = line.split('\\t')\n",
    "        \n",
    "        fh.write(lst[0]+'\\t'+'\\t'.join(lst[2:])+'\\n')\n",
    "        \n",
    "fh.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "methySite = OrderedDict()\n",
    "with open('methylation450_annotation.clean.txt') as f:\n",
    "    for line in f:\n",
    "        line = line.rstrip()\n",
    "        lst = line.split('\\t')\n",
    "        \n",
    "        methySite[lst[0]] ='\\t'.join(lst[1:])\n",
    "\n",
    "fh = open('Package/Summary/methylation450.univariate_cox.probe.annotation.TCGA-STAD.txt', 'wt')\n",
    "with open('Package/Summary/methylation450.univariate_cox.probe.TCGA-STAD.txt') as f:\n",
    "    for line in f:\n",
    "        line = line.rstrip()\n",
    "        lst = line.split('\\t')\n",
    "        \n",
    "        if line.startswith('coef'):\n",
    "            fh.write('\\t'+line+'\\t'+methySite['Composite Element REF']+'\\n')\n",
    "            continue\n",
    "            \n",
    "        fh.write(line+'\\t'+methySite[lst[0]]+'\\n')\n",
    "        \n",
    "        \n",
    "fh.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "####################################\n",
    "### for correlation analysis\n",
    "symbol2ensembl = OrderedDict()\n",
    "ensembl2symbol = OrderedDict()\n",
    "multiGene = OrderedDict()\n",
    "\n",
    "with open(database+'Homo_sapiens.GRCh38.86.gff3.clean.txt') as f:\n",
    "    for line in f:\n",
    "        line = line.rstrip()\n",
    "        lst = line.split('\\t')\n",
    "        if lst[1] not in symbol2ensembl.keys():\n",
    "            symbol2ensembl[lst[1]] = []\n",
    "        symbol2ensembl[lst[1]].append(lst[0].split('.')[0])\n",
    "        \n",
    "        if lst[0] not in ensembl2symbol.keys():\n",
    "            ensembl2symbol[lst[0].split('.')[0]] = lst[1]\n",
    "\n",
    "for ke, val in symbol2ensembl.items():\n",
    "    if len(val) > 1:\n",
    "        multiGene[ke] = val"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'WASH7P'"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ensembl2symbol['ENSG00000227232']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "RP11-944L7.5\n",
      "RP11-383H13.1\n",
      "RP11-383H13.1\n",
      "RP11-383H13.1\n",
      "RP11-383H13.1\n",
      "RP11-383H13.1\n",
      "RP11-383H13.1\n",
      "RP11-594N15.2\n",
      "AL589787.1\n",
      "AC021205.1\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "348"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "methyTarget = []\n",
    "i=0\n",
    "with open('methylation450.cox.probe_gender_age_stage.001.annotation.TCGA-STAD.txt') as f:\n",
    "    for line in f:\n",
    "        line.rstrip()\n",
    "        lst = line.split('\\t')\n",
    "        if lst[0] == '' or lst[6] == '*':\n",
    "            continue\n",
    "        genes = lst[9].split(';')\n",
    "        for gene in genes:\n",
    "            if gene not in symbol2ensembl.keys():\n",
    "                print (gene)\n",
    "                continue\n",
    "            pair = (lst[0], symbol2ensembl[gene][0])\n",
    "            if pair not in methyTarget:\n",
    "                \n",
    "                \n",
    "                \n",
    "                methyTarget.append(pair)\n",
    "                \n",
    "            ###### check if a gene has multiple ensembl ids\n",
    "            #if gene in multiGene.keys():\n",
    "            #    print (gene)\n",
    "            \n",
    "                \n",
    "                \n",
    "len(methyTarget)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "336\n"
     ]
    }
   ],
   "source": [
    "###################################\n",
    "##### overlapping samples\n",
    "\n",
    "from collections import OrderedDict\n",
    "\n",
    "methySamples = OrderedDict()\n",
    "ovlpSamples = []\n",
    "\n",
    "with open('methy450_STAD.txt') as f:\n",
    "    for line in f:\n",
    "        line = line.rstrip()\n",
    "        lst = line.split('\\t')\n",
    "        if lst[0].endswith('01A'):\n",
    "            methySamples[lst[0]] = lst[-1]\n",
    "\n",
    "with open('RNAseq_STAD.txt') as f:\n",
    "    for line in f:\n",
    "        line = line.rstrip()\n",
    "        lst = line.split('\\t')\n",
    "        if lst[0] in methySamples.keys():\n",
    "            ovlpSamples.append(lst[0])\n",
    "        \n",
    "\n",
    "print (len(ovlpSamples))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "methyDa = OrderedDict()\n",
    "rnaDa = OrderedDict()\n",
    "\n",
    "with open('methylation.beta_value450_and_traits.survival.TCGA-STAD.txt') as f:\n",
    "    for line in f:\n",
    "        line = line.rstrip()\n",
    "        lst = line.split('\\t')\n",
    "        if lst[0] == '':\n",
    "            samIndex1 = [lst.index(sam) for sam in ovlpSamples]\n",
    "            continue\n",
    "        \n",
    "        if lst[0] in [pair[0] for pair in methyTarget]:\n",
    "            methyDa[lst[0]] = [lst[i] for i in samIndex1]\n",
    "            \n",
    "\n",
    "with open('RNAseq_counts_data_TCGA_STAD.normalized.txt') as f:\n",
    "    for line in f:\n",
    "        line = line.rstrip()\n",
    "        lst = line.split('\\t')\n",
    "        if lst[0].startswith('TCGA'):\n",
    "            lst = [sam.replace('.', '-') for sam in lst]\n",
    "            samIndex2 = [lst.index(sam)+1 for sam in ovlpSamples]\n",
    "            continue\n",
    "            \n",
    "        gene = lst[0].split('.')[0]\n",
    "        \n",
    "        if gene in [pair[1] for pair in methyTarget]:\n",
    "            rnaDa[gene] = [lst[i] for i in samIndex2]\n",
    "            \n",
    "            \n",
    "fh1 = open('methylation450.beta.ovlp.TCGA-STAD.txt', 'wt')\n",
    "fh2 = open('RNAseq_counts.normalized.ovlp.TCGA-STAD.txt', 'wt')\n",
    "\n",
    "fh1.write('\\t'+'\\t'.join(ovlpSamples)+'\\n')\n",
    "fh2.write('\\t'+'\\t'.join(ovlpSamples)+'\\n')\n",
    "\n",
    "for probe, gene in methyTarget:\n",
    "    fh1.write(probe+'\\t'+'\\t'.join(methyDa[probe])+'\\n')\n",
    "    fh2.write(gene+'\\t'+'\\t'.join(rnaDa[gene])+'\\n')\n",
    "    \n",
    "fh1.close()\n",
    "fh2.close()\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "fh = open('correlation_methy_rna.sigSurvMethy.clean.TCGA-STAD.txt', 'wt')\n",
    "with open('correlation_methy_rna.sigSurvMethy.TCGA-STAD.txt') as f:\n",
    "    for line in f:\n",
    "        line = line.rstrip()\n",
    "        lst = line.split('\\t')\n",
    "        if lst[0] == 't':\n",
    "            fh.write('\\t'.join(['probe', 'symbol', 'geneID']+lst)+'\\n')\n",
    "            continue\n",
    "        pair = lst[0].split('-')\n",
    "        geneSymbol = [smbl for smbl in symbol2ensembl.keys() if [pair[1]]==symbol2ensembl[smbl]][0]\n",
    "        fh.write('\\t'.join([pair[0], geneSymbol, pair[1]]+lst[1:])+'\\n')\n",
    "        \n",
    "fh.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 96,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "##################################################\n",
    "########### DE Analysis between DM groups\n",
    "\n",
    "fh = open('DE_analysis_between_DM_groups.input.TCGA-STAD.txt', 'wt')\n",
    "\n",
    "probes = OrderedDict()\n",
    "with open('methylation450.beta.ovlp.TCGA-STAD.txt') as f:\n",
    "    for line in f:\n",
    "        line = line.rstrip()\n",
    "        lst = line.split('\\t')\n",
    "        if lst[0] == '':\n",
    "            fh.write(line+'\\n')\n",
    "            continue\n",
    "        elif len(probes) < 20:\n",
    "            probes[lst[0]] = lst[1:]\n",
    "            \n",
    "for ke, val in probes.items():\n",
    "    fh.write(ke+'\\t'+'\\t'.join(val)+'\\n')\n",
    "    \n",
    "\n",
    "with open('RNAseq.counts_only.TCGA-STAD.txt') as f:\n",
    "    for line in f:\n",
    "        line = line.rstrip()\n",
    "        lst = line.split('\\t')\n",
    "        if lst[0] == '':\n",
    "            samIndex = [lst.index(sam) for sam in ovlpSamples]\n",
    "            continue\n",
    "            \n",
    "        fh.write(lst[0].split('.')[0]+'\\t'+'\\t'.join([lst[i] for i in samIndex])+'\\n')\n",
    "\n",
    "fh.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 91,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "####################################################################\n",
    "############## survival inpput (hyper group, DEGs)\n",
    "\n",
    "sigCG = []\n",
    "with open('limma_001_2X.AKR1B1.methy_group.03.TCGA-BLCA.txt') as f:\n",
    "    for line in f:\n",
    "        line = line.rstrip()\n",
    "        lst = line.split('\\t')\n",
    "        if lst[0] == 'logFC':\n",
    "            continue\n",
    "        sigCG.append(lst[0])\n",
    "        \n",
    "         \n",
    "traits = ['sample_type','sample_type_id','tumor_stage','tumor_grade',\n",
    "          'age_at_diagnosis','days_to_death','days_to_last_follow_up',\n",
    "          'vital_status','race','gender','ethnicity']\n",
    "\n",
    "fh = open('methylation.beta_value450_and_traits.survival.TCGA-STAD.txt', 'wt')\n",
    "\n",
    "with open('methylation.beta_value450_and_traits.TCGA-STAD.txt') as f:\n",
    "    for line in f:\n",
    "        line = line.rstrip()\n",
    "        lst = line.split('\\t')\n",
    "        if lst[0] == '':\n",
    "            samples = [sam for sam in lst if re.search('-01A$', sam)]\n",
    "            samIndex = [lst.index(sam) for sam in samples]\n",
    "            fh.write('\\t'+'\\t'.join(samples)+'\\n')\n",
    "        elif lst[0] in traits:\n",
    "            trt = [lst[i] for i in samIndex]\n",
    "            fh.write(lst[0]+'\\t'+'\\t'.join(trt)+'\\n')\n",
    "        elif lst[0] in sigCG:\n",
    "            cgValue = [lst[i] for i in samIndex]\n",
    "            fh.write(lst[0]+'\\t'+'\\t'.join(cgValue)+'\\n')\n",
    "        \n",
    "\n",
    "fh.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 101,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "chemoTherapy = OrderedDict()\n",
    "with open('chemo_therapy.txt') as f:\n",
    "    for line in f:\n",
    "        line = line.rstrip()\n",
    "        lst = line.split('\\t')\n",
    "        \n",
    "        if lst[0] == '':\n",
    "            continue\n",
    "            \n",
    "        elif lst[1] == 'NA':\n",
    "            chemoTherapy[lst[0]+'-01A'] = 0\n",
    "            \n",
    "        else:\n",
    "            chemoTherapy[lst[0]+'-01A'] = 1\n",
    "            \n",
    "fh = open('chemo_therapy_info.input.txt', 'wt')\n",
    "with open('DE_analysis_between_DM_groups.input.TCGA-STAD.txt') as f:\n",
    "    for line in f:\n",
    "        line = line.rstrip()\n",
    "        lst = line.split('\\t')\n",
    "        \n",
    "        if lst[0] == '':\n",
    "            fh.write(line+'\\n')\n",
    "            chemo = [chemoTherapy[sam] for sam in lst if sam in chemoTherapy.keys()]\n",
    "            fh.write('chemo'+'\\t'+'\\t'.join(map(str,chemo))+'\\n')\n",
    "            break\n",
    "            \n",
    "fh.close()\n",
    "            "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "###############################################\n",
    "##### annotate degs\n",
    "\n",
    "gene = 'RP11-944L7.4'\n",
    "\n",
    "fh = open('Run/'+gene+'/limma_001_2X.'+gene+'.risk_group.symbol.TCGA-STAD.txt', 'wt')\n",
    "\n",
    "with open('Run/'+gene+'/limma_001_2X.'+gene+'.risk_group.TCGA-STAD.txt') as f:\n",
    "    for line in f:\n",
    "        line = line.rstrip()\n",
    "        lst = line.split('\\t')\n",
    "        if lst[0] == 'logFC':\n",
    "            fh.write('\\t'+'\\t'.join(['geneSymbol']+lst)+'\\n')\n",
    "            continue\n",
    "        elif lst[0] in ensembl2symbol.keys():\n",
    "            geneSymbol = ensembl2symbol[lst[0]]\n",
    "            fh.write('\\t'.join([lst[0],geneSymbol]+lst[1:])+'\\n')\n",
    "        else:\n",
    "            fh.write('\\t'.join([lst[0],lst[0]]+lst[1:])+'\\n')\n",
    "            \n",
    "fh.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "###############################################\n",
    "##### annotate degs\n",
    "\n",
    "gene = 'CLIP4'\n",
    "\n",
    "fh = open('Run/'+gene+'/newResults/limma_001_2X.'+gene+'.risk_group.symbol.TCGA-STAD.txt', 'wt')\n",
    "\n",
    "with open('Run/'+gene+'/newResults/limma_001_2X.'+gene+'.risk_group.TCGA-STAD.txt') as f:\n",
    "    for line in f:\n",
    "        line = line.rstrip()\n",
    "        lst = line.split('\\t')\n",
    "        if lst[0] == 'logFC':\n",
    "            fh.write('\\t'+'\\t'.join(['geneSymbol']+lst)+'\\n')\n",
    "            continue\n",
    "        elif lst[0] in ensembl2symbol.keys():\n",
    "            geneSymbol = ensembl2symbol[lst[0]]\n",
    "            fh.write('\\t'.join([lst[0],geneSymbol]+lst[1:])+'\\n')\n",
    "        else:\n",
    "            fh.write('\\t'.join([lst[0],lst[0]]+lst[1:])+'\\n')\n",
    "            \n",
    "fh.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ENSG00000155640\n",
      "ENSG00000188206\n",
      "ENSG00000205246\n",
      "ENSG00000213971\n",
      "ENSG00000225892\n",
      "ENSG00000226958\n",
      "ENSG00000229729\n",
      "ENSG00000231025\n",
      "ENSG00000241781\n",
      "ENSG00000244567\n",
      "ENSG00000261254\n",
      "ENSG00000269131\n",
      "ENSG00000269640\n",
      "ENSG00000269936\n",
      "ENSG00000272655\tPOLR2J4\n",
      "ENSG00000274735\n",
      "ENSG00000275530\n",
      "ENSG00000276924\n",
      "ENSG00000276967\n",
      "ENSG00000277105\n",
      "ENSG00000279978\n",
      "ENSG00000280111\n",
      "__no_feature\n",
      "__ambiguous\n",
      "__alignment_not_unique\n"
     ]
    }
   ],
   "source": [
    "################################################\n",
    "####### GSEA input\n",
    "\n",
    "gene = 'RP11-944L7.4'\n",
    "\n",
    "fh = open('Run/'+gene+'/'+gene+'.GSEA.expr.txt', 'wt')\n",
    "ovlpGene = OrderedDict()\n",
    "\n",
    "\n",
    "with open('Run/'+gene+'/'+gene+'.risk_group.genes_with_detectable_expression_levels.txt') as f:\n",
    "    for line in f:\n",
    "        line = line.rstrip()\n",
    "        lst = line.split('\\t')\n",
    "        if line.startswith('TCGA'):\n",
    "            fh.write('NAME\\t'+'\\t'.join(lst)+'\\n')\n",
    "            \n",
    "        elif lst[0] not in ensembl2symbol.keys():\n",
    "            print (lst[0])\n",
    "            continue\n",
    "            \n",
    "        else:\n",
    "            geneSymbol = ensembl2symbol[lst[0]]\n",
    "            if geneSymbol not in ovlpGene.keys():\n",
    "                ovlpGene[geneSymbol] = 1\n",
    "                fh.write(geneSymbol+'\\t'+'\\t'.join(lst[1:])+'\\n')\n",
    "                \n",
    "            else:\n",
    "                print (lst[0]+'\\t'+geneSymbol)\n",
    "        \n",
    "        \n",
    "fh.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "########################################\n",
    "###### validation dataset\n",
    "valClinical = OrderedDict()\n",
    "valExpr = OrderedDict()\n",
    "valMethy = OrderedDict()\n",
    "\n",
    "\n",
    "with open('validation/clinical.txt') as f:\n",
    "    for line in f:\n",
    "        line = line.rstrip()\n",
    "        lst = line.split('\\t')\n",
    "        valClinical[lst[1]] = lst[0]\n",
    "        \n",
    "with open('validation/expression.txt') as f:\n",
    "    for line in f:\n",
    "        line = line.rstrip()\n",
    "        lst = line.split('\\t')\n",
    "        for ke, val in valClinical.items():\n",
    "            if re.search(ke,lst[1]):\n",
    "                valExpr[ke] = lst[0]\n",
    "                \n",
    "with open('validation/methylation.txt') as f:\n",
    "    for line in f:\n",
    "        line = line.rstrip()\n",
    "        lst = line.split('\\t')\n",
    "        for ke, val in valExpr.items():\n",
    "            if re.search(ke+'$', lst[1]):\n",
    "                valMethy[ke] = lst[0]\n",
    "                \n",
    "                \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "ovlpValMethy = []\n",
    "ovlpValExpr = []\n",
    "ovlpValClinical = []\n",
    "\n",
    "\n",
    "for ke, val in valMethy.items():\n",
    "    ovlpValMethy.append(val)\n",
    "    ovlpValExpr.append(valExpr[ke])\n",
    "    ovlpValClinical.append(valClinical[ke])\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 88,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['GSM387920', 'GSM387912', 'GSM387914', 'GSM387934', 'GSM387924', 'GSM387933', 'GSM387921', 'GSM858124', 'GSM387918', 'GSM387929', 'GSM387935', 'GSM387932', 'GSM387915', 'GSM387916', 'GSM387925', 'GSM387930', 'GSM387913', 'GSM858121', 'GSM387928', 'GSM387927', 'GSM387919', 'GSM387908', 'GSM858125', 'GSM387815', 'GSM387881', 'GSM387898', 'GSM387931', 'GSM858110', 'GSM858119', 'GSM387922', 'GSM387893', 'GSM858123', 'GSM387899', 'GSM387897', 'GSM387900', 'GSM387885', 'GSM387878', 'GSM387944', 'GSM387971', 'GSM387966', 'GSM387950', 'GSM387980', 'GSM387945', 'GSM387970', 'GSM387823', 'GSM387838', 'GSM387826', 'GSM387824', 'GSM387836', 'GSM387831', 'GSM387835', 'GSM387832', 'GSM387819', 'GSM387817', 'GSM387983', 'GSM858088', 'GSM387862', 'GSM387987', 'GSM387822', 'GSM387982', 'GSM387834', 'GSM387833', 'GSM387848', 'GSM387854', 'GSM387842', 'GSM387829', 'GSM387821', 'GSM387859', 'GSM387864', 'GSM387850', 'GSM387828', 'GSM387825', 'GSM387839', 'GSM387985', 'GSM387820', 'GSM387837', 'GSM387827', 'GSM387882', 'GSM387904', 'GSM387973', 'GSM387874', 'GSM387871', 'GSM387868', 'GSM858133', 'GSM387872', 'GSM387949', 'GSM387877', 'GSM387911', 'GSM387953', 'GSM387895', 'GSM387936', 'GSM387962', 'GSM387942', 'GSM387959', 'GSM387961', 'GSM387905', 'GSM387889', 'GSM387814', 'GSM387978', 'GSM387952', 'GSM387967', 'GSM387981', 'GSM387818', 'GSM387943', 'GSM387940', 'GSM387954', 'GSM387955', 'GSM387986', 'GSM387951', 'GSM387977', 'GSM387958', 'GSM387816', 'GSM387906', 'GSM858128', 'GSM387858', 'GSM387979', 'GSM858127', 'GSM387946', 'GSM387963', 'GSM387852', 'GSM387855', 'GSM387907', 'GSM387846', 'GSM387860', 'GSM858117', 'GSM387857', 'GSM858114', 'GSM387841', 'GSM858090', 'GSM387851', 'GSM387939', 'GSM858113', 'GSM387976', 'GSM387880', 'GSM387901', 'GSM387849', 'GSM387861', 'GSM858103', 'GSM387870', 'GSM387876', 'GSM858130', 'GSM387890', 'GSM387896', 'GSM387894', 'GSM387909', 'GSM387883', 'GSM858118', 'GSM387888', 'GSM387903', 'GSM387902', 'GSM387891', 'GSM387879', 'GSM858105', 'GSM387892', 'GSM858131', 'GSM387948', 'GSM387947']\n"
     ]
    }
   ],
   "source": [
    "print (ovlpValExpr)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 95,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "fh = open('validation/GPL570-55999.clean.txt', 'wt')\n",
    "\n",
    "GPL96 = OrderedDict()\n",
    "GPL570 = OrderedDict()\n",
    "\n",
    "with open('validation/GPL96-57554.txt') as f:\n",
    "    for line in f:\n",
    "        line = line.rstrip()\n",
    "        lst = line.split('\\t')\n",
    "        if line.startswith('#') or lst[0]=='ID':\n",
    "            continue\n",
    "        if len(lst) < 10:\n",
    "            continue\n",
    "        GPL96[lst[0]] = lst[10].split(' /// ')\n",
    "        \n",
    "        \n",
    "with open('validation/GPL570-55999.txt') as f:\n",
    "    for line in f:\n",
    "        line = line.rstrip()\n",
    "        lst = line.split('\\t')\n",
    "        if line.startswith('#') or lst[0]=='ID':\n",
    "            continue\n",
    "        if len(lst) < 10:\n",
    "            continue\n",
    "            \n",
    "        GPL570[lst[0]] = lst[10].split(' /// ')\n",
    "        \n",
    "for ke ,val in GPL570.items():\n",
    "    fh.write('\\t'.join([ke]+val)+'\\n')\n",
    "    \n",
    "fh.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "fh1 = open('validation/methy.GSE30601_series_matrix.input.txt', 'wt')\n",
    "fh2 = open('validation/GSE15460-GPL96_series_matrix.input.txt', 'wt')\n",
    "fh3 = open('validation/GSE15460-GPL570_series_matrix.input.txt', 'wt')\n",
    "\n",
    "with open('validation/methy.GSE30601_series_matrix.txt') as f:\n",
    "    for line in f:\n",
    "        line = line.rstrip()\n",
    "        line = line.replace('\"', '')\n",
    "        lst = line.split('\\t')\n",
    "        \n",
    "        if line.startswith('!') or line == '':\n",
    "            continue\n",
    "            \n",
    "        elif line.startswith('ID_REF'):\n",
    "            samIndex = [lst.index(sam) for sam in ovlpValMethy if sam in lst]\n",
    "            fh1.write('\\t'+'\\t'.join([lst[i] for i in samIndex])+'\\n')\n",
    "        else:\n",
    "            fh1.write(lst[0]+'\\t'+'\\t'.join([lst[i] for i in samIndex])+'\\n')\n",
    "\n",
    "        \n",
    "with open('validation/GSE15460-GPL96_series_matrix.txt') as f:\n",
    "    for line in f:\n",
    "        line = line.rstrip()\n",
    "        line = line.replace('\"', '')\n",
    "        lst = line.split('\\t')\n",
    "        \n",
    "        if line.startswith('!') or line == '':\n",
    "            continue\n",
    "            \n",
    "        elif line.startswith('ID_REF'):\n",
    "            samIndex = [lst.index(sam) for sam in ovlpValExpr if sam in lst]\n",
    "            fh2.write('\\t'+'\\t'.join([lst[i] for i in samIndex])+'\\n')\n",
    "        else:\n",
    "            fh2.write(lst[0]+'\\t'+'\\t'.join([lst[i] for i in samIndex])+'\\n')\n",
    "\n",
    "        \n",
    "with open('validation/GSE15460-GPL570_series_matrix.txt') as f:\n",
    "    for line in f:\n",
    "        line = line.rstrip()\n",
    "        line = line.replace('\"', '')\n",
    "        lst = line.split('\\t')\n",
    "        \n",
    "        if line.startswith('!') or line == '':\n",
    "            continue\n",
    "            \n",
    "        elif line.startswith('ID_REF'):\n",
    "            samIndex = [lst.index(sam) for sam in ovlpValExpr if sam in lst]\n",
    "            fh3.write('\\t'+'\\t'.join([lst[i] for i in samIndex])+'\\n')\n",
    "        else:\n",
    "            fh3.write(lst[0]+'\\t'+'\\t'.join([lst[i] for i in samIndex])+'\\n')\n",
    "\n",
    "        \n",
    "fh1.close()\n",
    "fh2.close()\n",
    "fh3.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 93,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "fh4 = open('validation/GSE15460_clinical.input.txt', 'wt')\n",
    "\n",
    "with open('validation/GSE15460_clinical.txt') as f:\n",
    "    for line in f:\n",
    "        line = line.rstrip()\n",
    "        lst = line.split('\\t')\n",
    "        \n",
    "        if line.startswith('GSM'):\n",
    "            samIndex = [lst.index(sam) for sam in ovlpValClinical if sam in lst]\n",
    "            fh4.write('\\t'+'\\t'.join([lst[i] for i in samIndex])+'\\n')\n",
    "        elif line.startswith('OS') or line.startswith('Age_at_surgery'):\n",
    "            lst = [sam.split('.')[0] for sam in lst]\n",
    "            fh4.write(lst[0]+'\\t'+'\\t'.join([lst[i] for i in samIndex])+'\\n')\n",
    "        else:\n",
    "            fh4.write(lst[0]+'\\t'+'\\t'.join([lst[i] for i in samIndex])+'\\n')\n",
    "\n",
    "fh4.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "with open('methylation.beta_value27_and_traits.TCGA-STAD.txt') as f:\n",
    "    for line in f:\n",
    "        line = line.rstrip()\n",
    "        lst = line.split('\\t')\n",
    "        if lst[0] == '':\n",
    "            samples = [sam for sam in lst if re.search('-11A', sam)]\n",
    "            \n",
    "            \n",
    "            indexSamType = [lst.index(samtype) for samtype in lst if samtype == 'PrimaryTumor']\n",
    "            \n",
    "            \n",
    "        elif lst[0] == 'sample_type':\n",
    "            indexSamType = [lst.index(samtype) for samtype in lst if samtype == 'PrimaryTumor']\n",
    "            \n",
    "            fh.write(samples[0]+'\\t'+'\\t'.join([samples[i] for i in indexSamType])+'\\n')\n",
    "            \n",
    "        fh.write(lst[0]+'\\t'+'\\t'.join([lst[i] for i in indexSamType])+'\\n')\n",
    "        \n",
    "\n",
    "fh.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "fh = open('methylation.beta_value27_and_traits.TCGA-STAD.txt', 'wt')\n",
    "\n",
    "\n",
    "with open('methylation.beta_value27_and_traits.TCGA-STAD.txt') as f:\n",
    "    for line in f:\n",
    "        line = line.rstrip()\n",
    "        lst = line.split('\\t')\n",
    "        if lst[0] == '':\n",
    "            samples = [re.sub('-\\d+A', '', sam) for sam in lst]\n",
    "            \n",
    "        elif lst[0] == 'sample_type':\n",
    "            indexSamType = [lst.index(samtype) for samtype in lst if samtype == 'PrimaryTumor']\n",
    "            \n",
    "            fh.write(samples[0]+'\\t'+'\\t'.join([samples[i] for i in indexSamType])+'\\n')\n",
    "            \n",
    "        fh.write(lst[0]+'\\t'+'\\t'.join([lst[i] for i in indexSamType])+'\\n')\n",
    "        \n",
    "\n",
    "fh.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "###########################################\n",
    "### organize methylation annotation\n",
    "\n",
    "fh = open('methylation27_annotation.clean.txt', 'wt')\n",
    "with open('methylation27_annotation.txt') as f:\n",
    "    for line in f:\n",
    "        line = line.rstrip()\n",
    "        lst = line.split('\\t')\n",
    "        \n",
    "        fh.write(lst[0]+'\\t'+'\\t'.join(lst[2:])+'\\n')\n",
    "        \n",
    "fh.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "methySite = OrderedDict()\n",
    "with open('methylation27_annotation.clean.txt') as f:\n",
    "    for line in f:\n",
    "        line = line.rstrip()\n",
    "        lst = line.split('\\t')\n",
    "        \n",
    "        methySite[lst[0]] ='\\t'.join(lst[1:])\n",
    "\n",
    "fh = open('methylation27_test.annotation.TCGA-STAD.txt', 'wt')\n",
    "with open('methylation27_test.TCGA-STAD.txt') as f:\n",
    "    for line in f:\n",
    "        line = line.rstrip()\n",
    "        lst = line.split('\\t')\n",
    "        \n",
    "        if line.startswith('Tumor'):\n",
    "            fh.write('\\t'+line+'\\t'+methySite['Composite Element REF']+'\\n')\n",
    "            continue\n",
    "            \n",
    "        fh.write(line+'\\t'+methySite[lst[0]]+'\\n')\n",
    "        \n",
    "        \n",
    "fh.close()\n",
    "        "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "methy27:\n",
      "TCGA-BR-4187-01A\tPrimary Tumor\n",
      "TCGA-BR-4191-01A\tPrimary Tumor\n",
      "TCGA-BR-4201-01A\tPrimary Tumor\n",
      "TCGA-BR-4253-01A\tPrimary Tumor\n",
      "TCGA-BR-4256-01A\tPrimary Tumor\n",
      "TCGA-BR-4257-01A\tPrimary Tumor\n",
      "TCGA-BR-4267-01A\tPrimary Tumor\n",
      "TCGA-BR-4279-01A\tPrimary Tumor\n",
      "TCGA-BR-4280-01A\tPrimary Tumor\n",
      "TCGA-BR-4294-01A\tPrimary Tumor\n",
      "TCGA-BR-4357-01A\tPrimary Tumor\n",
      "TCGA-BR-4361-01A\tPrimary Tumor\n",
      "TCGA-BR-4363-01A\tPrimary Tumor\n",
      "TCGA-BR-4366-01A\tPrimary Tumor\n",
      "TCGA-BR-4367-01A\tPrimary Tumor\n",
      "TCGA-BR-4368-01A\tPrimary Tumor\n",
      "TCGA-BR-4369-01A\tPrimary Tumor\n",
      "TCGA-BR-4370-01A\tPrimary Tumor\n",
      "TCGA-BR-4371-01A\tPrimary Tumor\n",
      "TCGA-CG-4301-01A\tPrimary Tumor\n",
      "TCGA-CG-4304-01A\tPrimary Tumor\n",
      "TCGA-CG-4305-01A\tPrimary Tumor\n",
      "TCGA-CG-4306-01A\tPrimary Tumor\n",
      "TCGA-CG-4436-01A\tPrimary Tumor\n",
      "TCGA-CG-4438-01A\tPrimary Tumor\n",
      "TCGA-CG-4440-01A\tPrimary Tumor\n",
      "TCGA-CG-4442-01A\tPrimary Tumor\n",
      "TCGA-CG-4443-01A\tPrimary Tumor\n",
      "TCGA-CG-4444-01A\tPrimary Tumor\n",
      "TCGA-CG-4460-01A\tPrimary Tumor\n",
      "TCGA-CG-4462-01A\tPrimary Tumor\n",
      "TCGA-CG-4465-01A\tPrimary Tumor\n",
      "TCGA-CG-4466-01A\tPrimary Tumor\n",
      "TCGA-CG-4469-01A\tPrimary Tumor\n",
      "TCGA-CG-4475-01A\tPrimary Tumor\n",
      "TCGA-CG-4476-01A\tPrimary Tumor\n",
      "TCGA-CG-4477-01A\tPrimary Tumor\n",
      "methy450:\n",
      "TCGA-D7-A4YU-01A\tPrimary Tumor\n",
      "TCGA-3M-AB46-01A\tPrimary Tumor\n",
      "TCGA-3M-AB47-01A\tPrimary Tumor\n",
      "TCGA-B7-5818-01A\tPrimary Tumor\n",
      "TCGA-B7-A5TI-01A\tPrimary Tumor\n",
      "TCGA-B7-A5TJ-01A\tPrimary Tumor\n",
      "TCGA-B7-A5TK-01A\tPrimary Tumor\n",
      "TCGA-B7-A5TN-01A\tPrimary Tumor\n",
      "TCGA-BR-6452-01A\tPrimary Tumor\n",
      "TCGA-BR-6453-01A\tPrimary Tumor\n",
      "TCGA-BR-6454-01A\tPrimary Tumor\n",
      "TCGA-BR-6455-01A\tPrimary Tumor\n",
      "TCGA-BR-6456-01A\tPrimary Tumor\n",
      "TCGA-BR-6457-01A\tPrimary Tumor\n",
      "TCGA-BR-6458-01A\tPrimary Tumor\n",
      "TCGA-BR-6563-01A\tPrimary Tumor\n",
      "TCGA-BR-6564-01A\tPrimary Tumor\n",
      "TCGA-BR-6565-01A\tPrimary Tumor\n",
      "TCGA-BR-6566-01A\tPrimary Tumor\n",
      "TCGA-BR-6705-01A\tPrimary Tumor\n",
      "TCGA-BR-6707-01A\tPrimary Tumor\n",
      "TCGA-BR-6709-01A\tPrimary Tumor\n",
      "TCGA-BR-6710-01A\tPrimary Tumor\n",
      "TCGA-BR-6801-01A\tPrimary Tumor\n",
      "TCGA-BR-6802-01A\tPrimary Tumor\n",
      "TCGA-BR-6803-01A\tPrimary Tumor\n",
      "TCGA-BR-6852-01A\tPrimary Tumor\n",
      "TCGA-BR-7196-01A\tPrimary Tumor\n",
      "TCGA-BR-7197-01A\tPrimary Tumor\n",
      "TCGA-BR-7704-01A\tPrimary Tumor\n",
      "TCGA-BR-7707-01A\tPrimary Tumor\n",
      "TCGA-BR-7715-01A\tPrimary Tumor\n",
      "TCGA-BR-7716-01A\tPrimary Tumor\n",
      "TCGA-BR-7717-01A\tPrimary Tumor\n",
      "TCGA-BR-7722-01A\tPrimary Tumor\n",
      "TCGA-BR-7723-01A\tPrimary Tumor\n",
      "TCGA-BR-7851-01A\tPrimary Tumor\n",
      "TCGA-BR-7901-01A\tPrimary Tumor\n",
      "TCGA-BR-7957-01A\tPrimary Tumor\n",
      "TCGA-BR-7958-01A\tPrimary Tumor\n",
      "TCGA-BR-7959-01A\tPrimary Tumor\n",
      "TCGA-BR-8058-01A\tPrimary Tumor\n",
      "TCGA-BR-8059-01A\tPrimary Tumor\n",
      "TCGA-BR-8060-01A\tPrimary Tumor\n",
      "TCGA-BR-8077-01A\tPrimary Tumor\n",
      "TCGA-BR-8080-01A\tPrimary Tumor\n",
      "TCGA-BR-8081-01A\tPrimary Tumor\n",
      "TCGA-BR-8284-01A\tPrimary Tumor\n",
      "TCGA-BR-8286-01A\tPrimary Tumor\n",
      "TCGA-BR-8289-01A\tPrimary Tumor\n",
      "TCGA-BR-8291-01A\tPrimary Tumor\n",
      "TCGA-BR-8295-01A\tPrimary Tumor\n",
      "TCGA-BR-8296-01A\tPrimary Tumor\n",
      "TCGA-BR-8297-01A\tPrimary Tumor\n",
      "TCGA-BR-8361-01A\tPrimary Tumor\n",
      "TCGA-BR-8364-01A\tPrimary Tumor\n",
      "TCGA-BR-8365-01A\tPrimary Tumor\n",
      "TCGA-BR-8366-01A\tPrimary Tumor\n",
      "TCGA-BR-8367-01A\tPrimary Tumor\n",
      "TCGA-BR-8368-01A\tPrimary Tumor\n",
      "TCGA-BR-8369-01A\tPrimary Tumor\n",
      "TCGA-BR-8371-01A\tPrimary Tumor\n",
      "TCGA-BR-8372-01A\tPrimary Tumor\n",
      "TCGA-BR-8373-01A\tPrimary Tumor\n",
      "TCGA-BR-8380-01A\tPrimary Tumor\n",
      "TCGA-BR-8381-01A\tPrimary Tumor\n",
      "TCGA-BR-8382-01A\tPrimary Tumor\n",
      "TCGA-BR-8384-01A\tPrimary Tumor\n",
      "TCGA-BR-8483-01A\tPrimary Tumor\n",
      "TCGA-BR-8484-01A\tPrimary Tumor\n",
      "TCGA-BR-8485-01A\tPrimary Tumor\n",
      "TCGA-BR-8486-01A\tPrimary Tumor\n",
      "TCGA-BR-8487-01A\tPrimary Tumor\n",
      "TCGA-BR-8588-01A\tPrimary Tumor\n",
      "TCGA-BR-8589-01A\tPrimary Tumor\n",
      "TCGA-BR-8590-01A\tPrimary Tumor\n",
      "TCGA-BR-8591-01A\tPrimary Tumor\n",
      "TCGA-BR-8592-01A\tPrimary Tumor\n",
      "TCGA-BR-8676-01A\tPrimary Tumor\n",
      "TCGA-BR-8677-01A\tPrimary Tumor\n",
      "TCGA-BR-8678-01A\tPrimary Tumor\n",
      "TCGA-BR-8679-01A\tPrimary Tumor\n",
      "TCGA-BR-8680-01A\tPrimary Tumor\n",
      "TCGA-BR-8682-01A\tPrimary Tumor\n",
      "TCGA-BR-8683-01A\tPrimary Tumor\n",
      "TCGA-BR-8686-01A\tPrimary Tumor\n",
      "TCGA-BR-8687-01A\tPrimary Tumor\n",
      "TCGA-BR-8690-01A\tPrimary Tumor\n",
      "TCGA-BR-A44T-01A\tPrimary Tumor\n",
      "TCGA-BR-A44U-01A\tPrimary Tumor\n",
      "TCGA-BR-A4CR-01A\tPrimary Tumor\n",
      "TCGA-BR-A4CS-01A\tPrimary Tumor\n",
      "TCGA-BR-A4IV-01A\tPrimary Tumor\n",
      "TCGA-BR-A4J4-01A\tPrimary Tumor\n",
      "TCGA-BR-A4J5-01A\tPrimary Tumor\n",
      "TCGA-BR-A4J6-01A\tPrimary Tumor\n",
      "TCGA-BR-A4J7-01A\tPrimary Tumor\n",
      "TCGA-BR-A4J8-01A\tPrimary Tumor\n",
      "TCGA-BR-A4J9-01A\tPrimary Tumor\n",
      "TCGA-BR-A4PF-01A\tPrimary Tumor\n",
      "TCGA-BR-A4QL-01A\tPrimary Tumor\n",
      "TCGA-CD-5798-01A\tPrimary Tumor\n",
      "TCGA-CD-5799-01A\tPrimary Tumor\n",
      "TCGA-CD-5800-01A\tPrimary Tumor\n",
      "TCGA-CD-5801-01A\tPrimary Tumor\n",
      "TCGA-CD-5803-01A\tPrimary Tumor\n",
      "TCGA-CD-5804-01A\tPrimary Tumor\n",
      "TCGA-CD-5813-01A\tPrimary Tumor\n",
      "TCGA-CD-8524-01A\tPrimary Tumor\n",
      "TCGA-CD-8525-01A\tPrimary Tumor\n",
      "TCGA-CD-8526-01A\tPrimary Tumor\n",
      "TCGA-CD-8527-01A\tPrimary Tumor\n",
      "TCGA-CD-8528-01A\tPrimary Tumor\n",
      "TCGA-CD-8529-01A\tPrimary Tumor\n",
      "TCGA-CD-8530-01A\tPrimary Tumor\n",
      "TCGA-CD-8531-01A\tPrimary Tumor\n",
      "TCGA-CD-8532-01A\tPrimary Tumor\n",
      "TCGA-CD-8533-01A\tPrimary Tumor\n",
      "TCGA-CD-8534-01A\tPrimary Tumor\n",
      "TCGA-CD-8535-01A\tPrimary Tumor\n",
      "TCGA-CD-A486-01A\tPrimary Tumor\n",
      "TCGA-CD-A487-01A\tPrimary Tumor\n",
      "TCGA-CD-A489-01A\tPrimary Tumor\n",
      "TCGA-CD-A48A-01A\tPrimary Tumor\n",
      "TCGA-CD-A48C-01A\tPrimary Tumor\n",
      "TCGA-CD-A4MG-01A\tPrimary Tumor\n",
      "TCGA-CD-A4MH-01A\tPrimary Tumor\n",
      "TCGA-CG-4437-01A\tPrimary Tumor\n",
      "TCGA-CG-4441-01A\tPrimary Tumor\n",
      "TCGA-CG-5716-01A\tPrimary Tumor\n",
      "TCGA-CG-5717-01A\tPrimary Tumor\n",
      "TCGA-CG-5718-01A\tPrimary Tumor\n",
      "TCGA-CG-5719-01A\tPrimary Tumor\n",
      "TCGA-CG-5720-01A\tPrimary Tumor\n",
      "TCGA-CG-5721-01A\tPrimary Tumor\n",
      "TCGA-CG-5722-01A\tPrimary Tumor\n",
      "TCGA-CG-5723-01A\tPrimary Tumor\n",
      "TCGA-CG-5724-01A\tPrimary Tumor\n",
      "TCGA-CG-5725-01A\tPrimary Tumor\n",
      "TCGA-CG-5726-01A\tPrimary Tumor\n",
      "TCGA-CG-5732-01A\tPrimary Tumor\n",
      "TCGA-CG-5734-01A\tPrimary Tumor\n",
      "TCGA-D7-5577-01A\tPrimary Tumor\n",
      "TCGA-D7-5578-01A\tPrimary Tumor\n",
      "TCGA-D7-6519-01A\tPrimary Tumor\n",
      "TCGA-D7-6520-01A\tPrimary Tumor\n",
      "TCGA-D7-6521-01A\tPrimary Tumor\n",
      "TCGA-D7-6522-01A\tPrimary Tumor\n",
      "TCGA-D7-6524-01A\tPrimary Tumor\n",
      "TCGA-D7-6525-01A\tPrimary Tumor\n",
      "TCGA-D7-6526-01A\tPrimary Tumor\n",
      "TCGA-D7-6527-01A\tPrimary Tumor\n",
      "TCGA-D7-6528-01A\tPrimary Tumor\n",
      "TCGA-D7-6815-01A\tPrimary Tumor\n",
      "TCGA-D7-6818-01A\tPrimary Tumor\n",
      "TCGA-D7-6822-01A\tPrimary Tumor\n",
      "TCGA-D7-8570-01A\tPrimary Tumor\n",
      "TCGA-D7-8572-01A\tPrimary Tumor\n",
      "TCGA-D7-8573-01A\tPrimary Tumor\n",
      "TCGA-D7-8574-01A\tPrimary Tumor\n",
      "TCGA-D7-8575-01A\tPrimary Tumor\n",
      "TCGA-D7-8576-01A\tPrimary Tumor\n",
      "TCGA-D7-8578-01A\tPrimary Tumor\n",
      "TCGA-D7-8579-01A\tPrimary Tumor\n",
      "TCGA-D7-A4YX-01A\tPrimary Tumor\n",
      "TCGA-D7-A4Z0-01A\tPrimary Tumor\n",
      "TCGA-D7-A6EV-01A\tPrimary Tumor\n",
      "TCGA-D7-A6EX-01A\tPrimary Tumor\n",
      "TCGA-D7-A6EY-01A\tPrimary Tumor\n",
      "TCGA-D7-A6EZ-01A\tPrimary Tumor\n",
      "TCGA-D7-A6F0-01A\tPrimary Tumor\n",
      "TCGA-D7-A6F2-01A\tPrimary Tumor\n",
      "TCGA-D7-A747-01A\tPrimary Tumor\n",
      "TCGA-D7-A748-01A\tPrimary Tumor\n",
      "TCGA-D7-A74A-01A\tPrimary Tumor\n",
      "TCGA-EQ-8122-01A\tPrimary Tumor\n",
      "TCGA-F1-6177-01A\tPrimary Tumor\n",
      "TCGA-F1-6874-01A\tPrimary Tumor\n",
      "TCGA-F1-6875-01A\tPrimary Tumor\n",
      "TCGA-F1-A448-01A\tPrimary Tumor\n",
      "TCGA-F1-A72C-01A\tPrimary Tumor\n",
      "TCGA-FP-7735-01A\tPrimary Tumor\n",
      "TCGA-FP-7829-01A\tPrimary Tumor\n",
      "TCGA-FP-7916-01A\tPrimary Tumor\n",
      "TCGA-FP-7998-01A\tPrimary Tumor\n",
      "TCGA-FP-8099-01A\tPrimary Tumor\n",
      "TCGA-FP-8209-01A\tPrimary Tumor\n",
      "TCGA-FP-8210-01A\tPrimary Tumor\n",
      "TCGA-FP-8211-01A\tPrimary Tumor\n",
      "TCGA-FP-8631-01A\tPrimary Tumor\n",
      "TCGA-FP-A4BF-01A\tPrimary Tumor\n",
      "TCGA-FP-A8CX-01A\tPrimary Tumor\n",
      "TCGA-FP-A9TM-01A\tPrimary Tumor\n",
      "TCGA-HF-7131-01A\tPrimary Tumor\n",
      "TCGA-HF-7132-01A\tPrimary Tumor\n",
      "TCGA-HF-7133-01A\tPrimary Tumor\n",
      "TCGA-HF-7134-01A\tPrimary Tumor\n",
      "TCGA-HF-A5NB-01A\tPrimary Tumor\n",
      "TCGA-HJ-7597-01A\tPrimary Tumor\n",
      "TCGA-HU-8238-01A\tPrimary Tumor\n",
      "TCGA-HU-8244-01A\tPrimary Tumor\n",
      "TCGA-HU-8249-01A\tPrimary Tumor\n",
      "TCGA-HU-8602-01A\tPrimary Tumor\n",
      "TCGA-HU-8604-01A\tPrimary Tumor\n",
      "TCGA-HU-8608-01A\tPrimary Tumor\n",
      "TCGA-HU-8610-01A\tPrimary Tumor\n",
      "TCGA-HU-A4G2-01A\tPrimary Tumor\n",
      "TCGA-HU-A4G3-01A\tPrimary Tumor\n",
      "TCGA-HU-A4G8-01A\tPrimary Tumor\n",
      "TCGA-HU-A4G9-01A\tPrimary Tumor\n",
      "TCGA-HU-A4GC-01A\tPrimary Tumor\n",
      "TCGA-HU-A4GD-01A\tPrimary Tumor\n",
      "TCGA-HU-A4GF-01A\tPrimary Tumor\n",
      "TCGA-HU-A4GH-01A\tPrimary Tumor\n",
      "TCGA-HU-A4GJ-01A\tPrimary Tumor\n",
      "TCGA-HU-A4GP-01A\tPrimary Tumor\n",
      "TCGA-HU-A4GQ-01A\tPrimary Tumor\n",
      "TCGA-HU-A4GT-01A\tPrimary Tumor\n",
      "TCGA-HU-A4GU-01A\tPrimary Tumor\n",
      "TCGA-HU-A4GX-01A\tPrimary Tumor\n",
      "TCGA-HU-A4GY-01A\tPrimary Tumor\n",
      "TCGA-HU-A4H0-01A\tPrimary Tumor\n",
      "TCGA-HU-A4H2-01A\tPrimary Tumor\n",
      "TCGA-HU-A4H3-01A\tPrimary Tumor\n",
      "TCGA-HU-A4H4-01A\tPrimary Tumor\n",
      "TCGA-HU-A4H5-01A\tPrimary Tumor\n",
      "TCGA-HU-A4H6-01A\tPrimary Tumor\n",
      "TCGA-HU-A4H8-01A\tPrimary Tumor\n",
      "TCGA-HU-A4HB-01A\tPrimary Tumor\n",
      "TCGA-HU-A4HD-01A\tPrimary Tumor\n",
      "TCGA-IN-7806-01A\tPrimary Tumor\n",
      "TCGA-IN-7808-01A\tPrimary Tumor\n",
      "TCGA-IN-8462-01A\tPrimary Tumor\n",
      "TCGA-IN-8663-01A\tPrimary Tumor\n",
      "TCGA-IN-A6RI-01A\tPrimary Tumor\n",
      "TCGA-IN-A6RJ-01A\tPrimary Tumor\n",
      "TCGA-IN-A6RL-01A\tPrimary Tumor\n",
      "TCGA-IN-A6RN-01A\tPrimary Tumor\n",
      "TCGA-IN-A6RO-01A\tPrimary Tumor\n",
      "TCGA-IN-A6RR-01A\tPrimary Tumor\n",
      "TCGA-IN-A6RS-01A\tPrimary Tumor\n",
      "TCGA-IN-A7NR-01A\tPrimary Tumor\n",
      "TCGA-IN-A7NT-01A\tPrimary Tumor\n",
      "TCGA-IN-A7NU-01A\tPrimary Tumor\n",
      "TCGA-IN-AB1V-01A\tPrimary Tumor\n",
      "TCGA-IN-AB1X-01A\tPrimary Tumor\n",
      "TCGA-IP-7968-01A\tPrimary Tumor\n",
      "TCGA-KB-A6F7-01A\tPrimary Tumor\n",
      "TCGA-KB-A93G-01A\tPrimary Tumor\n",
      "TCGA-KB-A93H-01A\tPrimary Tumor\n",
      "TCGA-KB-A93J-01A\tPrimary Tumor\n",
      "TCGA-MX-A5UG-01A\tPrimary Tumor\n",
      "TCGA-MX-A5UJ-01A\tPrimary Tumor\n",
      "TCGA-MX-A663-01A\tPrimary Tumor\n",
      "TCGA-MX-A666-01A\tPrimary Tumor\n",
      "TCGA-R5-A7O7-01A\tPrimary Tumor\n",
      "TCGA-R5-A7ZE-01B\tPrimary Tumor\n",
      "TCGA-R5-A7ZF-01A\tPrimary Tumor\n",
      "TCGA-R5-A7ZI-01A\tPrimary Tumor\n",
      "TCGA-R5-A7ZR-01A\tPrimary Tumor\n",
      "TCGA-R5-A805-01A\tPrimary Tumor\n",
      "TCGA-RD-A7BS-01A\tPrimary Tumor\n",
      "TCGA-RD-A7BT-01A\tPrimary Tumor\n",
      "TCGA-RD-A7BW-01A\tPrimary Tumor\n",
      "TCGA-RD-A7C1-01A\tPrimary Tumor\n",
      "TCGA-RD-A8MV-01A\tPrimary Tumor\n",
      "TCGA-RD-A8MW-01A\tPrimary Tumor\n",
      "TCGA-RD-A8N0-01A\tPrimary Tumor\n",
      "TCGA-RD-A8N1-01A\tPrimary Tumor\n",
      "TCGA-RD-A8N2-01A\tPrimary Tumor\n",
      "TCGA-RD-A8N4-01A\tPrimary Tumor\n",
      "TCGA-RD-A8N5-01A\tPrimary Tumor\n",
      "TCGA-RD-A8N6-01A\tPrimary Tumor\n",
      "TCGA-RD-A8N9-01A\tPrimary Tumor\n",
      "TCGA-RD-A8NB-01A\tPrimary Tumor\n",
      "TCGA-SW-A7EA-01A\tPrimary Tumor\n",
      "TCGA-SW-A7EB-01A\tPrimary Tumor\n",
      "TCGA-VQ-A8DT-01A\tPrimary Tumor\n",
      "TCGA-VQ-A8DU-01A\tPrimary Tumor\n",
      "TCGA-VQ-A8DV-01A\tPrimary Tumor\n",
      "TCGA-VQ-A8DZ-01A\tPrimary Tumor\n",
      "TCGA-VQ-A8E0-01A\tPrimary Tumor\n",
      "TCGA-VQ-A8E2-01A\tPrimary Tumor\n",
      "TCGA-VQ-A8E3-01A\tPrimary Tumor\n",
      "TCGA-VQ-A8E7-01B\tPrimary Tumor\n",
      "TCGA-VQ-A8P2-01A\tPrimary Tumor\n",
      "TCGA-VQ-A8P3-01A\tPrimary Tumor\n",
      "TCGA-VQ-A8P5-01A\tPrimary Tumor\n",
      "TCGA-VQ-A8P8-01A\tPrimary Tumor\n",
      "TCGA-VQ-A8PB-01A\tPrimary Tumor\n",
      "TCGA-VQ-A8PC-01A\tPrimary Tumor\n",
      "TCGA-VQ-A8PD-01A\tPrimary Tumor\n",
      "TCGA-VQ-A8PE-01A\tPrimary Tumor\n",
      "TCGA-VQ-A8PF-01A\tPrimary Tumor\n",
      "TCGA-VQ-A8PH-01A\tPrimary Tumor\n",
      "TCGA-VQ-A8PJ-01A\tPrimary Tumor\n",
      "TCGA-VQ-A8PK-01A\tPrimary Tumor\n",
      "TCGA-VQ-A8PM-01A\tPrimary Tumor\n",
      "TCGA-VQ-A8PO-01A\tPrimary Tumor\n",
      "TCGA-VQ-A8PP-01A\tPrimary Tumor\n",
      "TCGA-VQ-A8PQ-01A\tPrimary Tumor\n",
      "TCGA-VQ-A8PU-01A\tPrimary Tumor\n",
      "TCGA-VQ-A8PX-01A\tPrimary Tumor\n",
      "TCGA-VQ-A91A-01A\tPrimary Tumor\n",
      "TCGA-VQ-A91D-01A\tPrimary Tumor\n",
      "TCGA-VQ-A91E-01A\tPrimary Tumor\n",
      "TCGA-VQ-A91K-01A\tPrimary Tumor\n",
      "TCGA-VQ-A91N-01A\tPrimary Tumor\n",
      "TCGA-VQ-A91Q-01A\tPrimary Tumor\n",
      "TCGA-VQ-A91S-01A\tPrimary Tumor\n",
      "TCGA-VQ-A91U-01A\tPrimary Tumor\n",
      "TCGA-VQ-A91V-01A\tPrimary Tumor\n",
      "TCGA-VQ-A91X-01A\tPrimary Tumor\n",
      "TCGA-VQ-A91Y-01A\tPrimary Tumor\n",
      "TCGA-VQ-A91Z-01A\tPrimary Tumor\n",
      "TCGA-VQ-A922-01A\tPrimary Tumor\n",
      "TCGA-VQ-A923-01A\tPrimary Tumor\n",
      "TCGA-VQ-A924-01A\tPrimary Tumor\n",
      "TCGA-VQ-A925-01A\tPrimary Tumor\n",
      "TCGA-VQ-A927-01A\tPrimary Tumor\n",
      "TCGA-VQ-A928-01A\tPrimary Tumor\n",
      "TCGA-VQ-A92D-01A\tPrimary Tumor\n",
      "TCGA-VQ-A94O-01A\tPrimary Tumor\n",
      "TCGA-VQ-A94P-01A\tPrimary Tumor\n",
      "TCGA-VQ-A94R-01A\tPrimary Tumor\n",
      "TCGA-VQ-A94T-01A\tPrimary Tumor\n",
      "TCGA-VQ-A94U-01A\tPrimary Tumor\n",
      "TCGA-VQ-AA64-01A\tPrimary Tumor\n",
      "TCGA-VQ-AA68-01A\tPrimary Tumor\n",
      "TCGA-VQ-AA69-01A\tPrimary Tumor\n",
      "TCGA-VQ-AA6A-01A\tPrimary Tumor\n",
      "TCGA-VQ-AA6D-01A\tPrimary Tumor\n",
      "TCGA-VQ-AA6F-01A\tPrimary Tumor\n",
      "TCGA-VQ-AA6G-01A\tPrimary Tumor\n",
      "TCGA-VQ-AA6J-01A\tPrimary Tumor\n",
      "TCGA-VQ-AA6K-01A\tPrimary Tumor\n",
      "TCGA-ZA-A8F6-01A\tPrimary Tumor\n",
      "TCGA-ZQ-A9CR-01A\tPrimary Tumor\n",
      "338\n"
     ]
    }
   ],
   "source": [
    "from collections import OrderedDict\n",
    "\n",
    "methy = OrderedDict()\n",
    "with open('RNAseq_STAD.txt') as f:\n",
    "    for line in f:\n",
    "        line = line.rstrip()\n",
    "        lst = line.split('\\t')\n",
    "        methy[lst[0]] = lst[1]\n",
    "        \n",
    "        \n",
    "print ('methy27:')\n",
    "with open('methy27_STAD.txt') as f:\n",
    "    for line in f:\n",
    "        line = line.rstrip()\n",
    "        lst = line.split('\\t')\n",
    "        if lst[0] in methy.keys():\n",
    "            print (line)\n",
    "        \n",
    "\n",
    "print ('methy450:')\n",
    "x=0\n",
    "with open('methy450_STAD.txt') as f:\n",
    "    for line in f:\n",
    "        line = line.rstrip()\n",
    "        lst = line.split('\\t')\n",
    "        if lst[0] in methy.keys():\n",
    "            print (line)\n",
    "            x+=1\n",
    "            \n",
    "print (x)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "anaconda-cloud": {},
  "kernelspec": {
   "display_name": "Python [Root]",
   "language": "python",
   "name": "Python [Root]"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.5.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}
