{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "import os\n",
    "import re\n",
    "from collections import OrderedDict\n",
    "\n",
    "os.chdir('/home/ruidong/Documents/Research/share/miRNA/new_analysis/')\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "diffMIR = OrderedDict()\n",
    "allMIR = OrderedDict()\n",
    "with open('diffmir.txt') as f:\n",
    "    for line in f:\n",
    "        line = line.rstrip()\n",
    "        lst = line.split('\\t')\n",
    "        if lst[0] == 'miR-name':\n",
    "            continue\n",
    "        else:\n",
    "            diffMIR[lst[0]] = lst[-1]\n",
    "            \n",
    "            \n",
    "fh1 = open('allmirseq.clean.txt', 'wt')\n",
    "fh2 = open('all_mir.fa', 'wt')\n",
    "fh3 = open('diff_mir.fa', 'wt')\n",
    "\n",
    "test = OrderedDict()        \n",
    "with open('allmirseq.txt') as f:\n",
    "    for line in f:\n",
    "        line = line.rstrip()\n",
    "        lst= line.split('\\t')\n",
    "        if lst[0] == 'miRNA_name':\n",
    "            fh1.write(line+'\\n')\n",
    "            continue\n",
    "        elif lst[0] in allMIR.keys():\n",
    "            continue\n",
    "        \n",
    "        allMIR[lst[0]] = lst[1]\n",
    "        fh1.write(line+'\\n')\n",
    "        \n",
    "        fh2.write('>'+lst[0]+'\\n'+lst[1]+'\\n')\n",
    "        \n",
    "        if lst[0] in diffMIR.keys():\n",
    "            test[lst[0]] = 1\n",
    "            fh3.write('>'+lst[0]+'\\n'+lst[1]+'\\n')\n",
    "            \n",
    "            \n",
    "fh1.close()\n",
    "fh2.close()\n",
    "fh3.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "494\n"
     ]
    }
   ],
   "source": [
    "diffGENE = OrderedDict()\n",
    "with open('diffmrna.txt') as f:\n",
    "    for line in f:\n",
    "        line = line.rstrip()\n",
    "        lst = line.split('\\t')\n",
    "        \n",
    "        if line.startswith('gene_id'):\n",
    "            continue\n",
    "        diffGENE[lst[0]] = lst[8]\n",
    "\n",
    "        \n",
    "geneSEQ = OrderedDict()\n",
    "fh = open('Gmax_275_Wm82.a2.v1.transcript.diff.fa', 'wt')\n",
    "with open('Gmax_275_Wm82.a2.v1.transcript.fa') as f:\n",
    "    for line in f:\n",
    "        if line.startswith('>'):\n",
    "            trpt = line\n",
    "            mch = re.search('locus=(.+) ID', line)\n",
    "            gene = mch.group(1)\n",
    "            if gene in diffGENE.keys():\n",
    "                geneSEQ[trpt] = ''\n",
    "                fh.write(trpt)\n",
    "                continue\n",
    "                \n",
    "        if trpt in geneSEQ.keys():   \n",
    "            fh.write(line.upper())\n",
    "            #geneSEQ[trpt] += line\n",
    "                \n",
    "fh.close()\n",
    "                \n",
    "print (len(geneSEQ))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 149,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "number of interactions in degradome: 3373\n",
      "number of interactions in pstarget: 791\n",
      "number of interactions in overlaps: 251\n"
     ]
    }
   ],
   "source": [
    "degradome= []\n",
    "pstarget = []\n",
    "overlaps = []\n",
    "\n",
    "\n",
    "fh1 = open('degradome_target.clean.txt', 'wt')\n",
    "with open('dedradome_target.txt') as f:\n",
    "    for line in f:\n",
    "        if line.startswith('miRNA'):\n",
    "            continue\n",
    "        line = line.rstrip()\n",
    "        lst = line.split('\\t')\n",
    "        \n",
    "        lst[1] = re.sub(r'(\\d+)g', r'\\1G', lst[1])\n",
    "        \n",
    "        miRNAs = re.split(',| ', lst[0])\n",
    "        \n",
    "        for mir in miRNAs:\n",
    "            if mir == '':\n",
    "                continue\n",
    "\n",
    "            if mir+'\\t'+lst[1] not in degradome:\n",
    "                degradome.append(mir+'\\t'+lst[1])\n",
    "                fh1.write(mir+'\\t'+lst[1]+'\\n')\n",
    "                \n",
    "fh1.close()\n",
    "\n",
    "        \n",
    "                \n",
    "with open('psRNATarget_diffmir_JGI.txt') as f:\n",
    "    for line in f:\n",
    "        if line.startswith('#') or line.startswith('miRNA'):\n",
    "            continue\n",
    "        line = line.rstrip()\n",
    "        lst = line.split('\\t')\n",
    "       \n",
    "        if lst[0]+'\\t'+lst[1] not in pstarget:\n",
    "            pstarget.append(lst[0]+'\\t'+lst[1])\n",
    "\n",
    "                      \n",
    "fh = open('degradome_pstarget_overlaps.txt', 'wt')\n",
    "for target in degradome:\n",
    "    if target in pstarget:\n",
    "        overlaps.append(target)\n",
    "        fh.write(target+'\\n')\n",
    "        \n",
    "        \n",
    "fh.close()\n",
    "        \n",
    "        \n",
    "print ('number of interactions in %s: %d' %('degradome', len(degradome)))\n",
    "print ('number of interactions in %s: %d' %('pstarget', len(pstarget)))\n",
    "\n",
    "print ('number of interactions in %s: %d' %('overlaps', len(overlaps)))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 150,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "degradome = OrderedDict()\n",
    "\n",
    "\n",
    "fh = open('degradome_target.diffmir.clean.txt', 'wt')\n",
    "with open('degradome_target.clean.txt') as f:\n",
    "    for line in f:\n",
    "        line = line.rstrip()\n",
    "        lst = line.split('\\t')\n",
    "        \n",
    "        \n",
    "        if lst[0] not in diffMIR.keys():\n",
    "            continue\n",
    "        \n",
    "        elif line not in degradome.keys():\n",
    "            fh.write(line+'\\n')\n",
    "            degradome[line] = 1\n",
    "            \n",
    "fh.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 151,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "number of miRNAs and targets in degradome: 226, 460\n",
      "number of miRNAs and targets in psRNAtarget: 74, 361\n",
      "number of miRNAs and targets in overlaps: 22, 93\n"
     ]
    }
   ],
   "source": [
    "deg_mir = []\n",
    "deg_target = []\n",
    "ps_mir = []\n",
    "ps_target = []\n",
    "ovlp_mir = []\n",
    "ovlp_target = []\n",
    "\n",
    "degradome= OrderedDict()\n",
    "pstarget = OrderedDict()\n",
    "overlaps = OrderedDict()\n",
    "union = OrderedDict()\n",
    "\n",
    "with open('degradome_target.clean.txt') as f:\n",
    "    for line in f:\n",
    "    \n",
    "        line = line.rstrip()\n",
    "        lst = line.split('\\t')\n",
    "        \n",
    "        if lst[0] not in deg_mir:\n",
    "            deg_mir.append(lst[0])\n",
    "            \n",
    "        if lst[1] not in deg_target:\n",
    "            deg_target.append(lst[1])\n",
    "            \n",
    "        if lst[0] not in degradome.keys():\n",
    "            degradome[lst[0]] = []\n",
    "            \n",
    "        degradome[lst[0]].append(lst[1])\n",
    "            \n",
    "            \n",
    "print ('number of miRNAs and targets in %s: %d, %d' %('degradome', len(deg_mir), len(deg_target)))\n",
    "\n",
    "\n",
    "with open('psRNATarget_diffmir_JGI.txt') as f:\n",
    "    for line in f:\n",
    "        if line.startswith('#') or line.startswith('miRNA'):\n",
    "            continue\n",
    "            \n",
    "        line = line.rstrip()\n",
    "        lst = line.split('\\t')\n",
    "        \n",
    "        if lst[0] not in ps_mir:\n",
    "            ps_mir.append(lst[0])\n",
    "            \n",
    "        if lst[1] not in ps_target:\n",
    "            ps_target.append(lst[1])\n",
    "            \n",
    "            \n",
    "        if lst[0] not in pstarget.keys():\n",
    "            pstarget[lst[0]] = []\n",
    "            \n",
    "        pstarget[lst[0]].append(lst[1])\n",
    "            \n",
    "            \n",
    "print ('number of miRNAs and targets in %s: %d, %d' %('psRNAtarget', len(ps_mir), len(ps_target)))\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "with open('degradome_pstarget_overlaps.txt') as f:\n",
    "    for line in f:\n",
    "        line = line.rstrip()\n",
    "        lst = line.split('\\t')\n",
    "        \n",
    "        if lst[0] not in ovlp_mir:\n",
    "            ovlp_mir.append(lst[0])\n",
    "            \n",
    "        if lst[1] not in ovlp_target:\n",
    "            ovlp_target.append(lst[1])\n",
    "            \n",
    "            \n",
    "        if lst[0] not in overlaps.keys():\n",
    "            overlaps[lst[0]] = []\n",
    "            \n",
    "        overlaps[lst[0]].append(lst[1])\n",
    "        \n",
    "print ('number of miRNAs and targets in %s: %d, %d' %('overlaps', len(ovlp_mir), len(ovlp_target)))\n",
    "\n",
    "\n",
    "\n",
    "fh1 = open('degradome_target.organized.txt', 'wt')\n",
    "fh2 = open('psRNATarget_diffmir_JGI.organized.txt', 'wt')\n",
    "fh3 = open('degradome_pstarget_overlaps.organized.txt', 'wt')\n",
    "fh4 = open('degradome_target.diffmir.organized.txt', 'wt')\n",
    "\n",
    "for ke, val in degradome.items():\n",
    "    fh1.write(ke+'\\t'+';'.join(val)+'\\n')\n",
    "    \n",
    "    if ke in diffMIR.keys():\n",
    "        fh4.write(ke+'\\t'+';'.join(val)+'\\n')\n",
    "    \n",
    "for ke, val in pstarget.items():\n",
    "    fh2.write(ke+'\\t'+';'.join(val)+'\\n')\n",
    "    \n",
    "for ke, val in overlaps.items():\n",
    "    fh3.write(ke+'\\t'+';'.join(val)+'\\n')\n",
    "    \n",
    "fh1.close()\n",
    "fh2.close()\n",
    "fh3.close()\n",
    "fh4.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "number of miRNAs and targets in degradome: 226, 460\n",
      "number of miRNAs and targets in psRNAtarget: 74, 361\n",
      "number of miRNAs and targets in overlaps: 22, 93\n"
     ]
    }
   ],
   "source": [
    "deg_mir = []\n",
    "deg_target = []\n",
    "ps_mir = []\n",
    "ps_target = []\n",
    "ovlp_mir = []\n",
    "ovlp_target = []\n",
    "\n",
    "degradome= OrderedDict()\n",
    "pstarget = OrderedDict()\n",
    "overlaps = OrderedDict()\n",
    "union = OrderedDict()\n",
    "\n",
    "with open('degradome_target.clean.txt') as f:\n",
    "    for line in f:\n",
    "    \n",
    "        line = line.rstrip()\n",
    "        lst = line.split('\\t')\n",
    "        \n",
    "        if lst[0] not in deg_mir:\n",
    "            deg_mir.append(lst[0])\n",
    "            \n",
    "        if lst[1] not in deg_target:\n",
    "            deg_target.append(lst[1])\n",
    "            \n",
    "        if lst[0] not in degradome.keys():\n",
    "            degradome[lst[0]] = []\n",
    "            \n",
    "        xxx = lst[1].split('.')[0]+'.'+lst[1].split('.')[1]\n",
    "        if xxx not in degradome[lst[0]]:\n",
    "            degradome[lst[0]].append(xxx)\n",
    "            \n",
    "            \n",
    "print ('number of miRNAs and targets in %s: %d, %d' %('degradome', len(deg_mir), len(deg_target)))\n",
    "\n",
    "\n",
    "with open('psRNATarget_diffmir_JGI.txt') as f:\n",
    "    for line in f:\n",
    "        if line.startswith('#') or line.startswith('miRNA'):\n",
    "            continue\n",
    "            \n",
    "        line = line.rstrip()\n",
    "        lst = line.split('\\t')\n",
    "        \n",
    "        if lst[0] not in ps_mir:\n",
    "            ps_mir.append(lst[0])\n",
    "            \n",
    "        if lst[1] not in ps_target:\n",
    "            ps_target.append(lst[1])\n",
    "            \n",
    "            \n",
    "        if lst[0] not in pstarget.keys():\n",
    "            pstarget[lst[0]] = []\n",
    "            \n",
    "        xxx = lst[1].split('.')[0]+'.'+lst[1].split('.')[1]\n",
    "        if xxx not in pstarget[lst[0]]:\n",
    "            pstarget[lst[0]].append(xxx)\n",
    "            \n",
    "print ('number of miRNAs and targets in %s: %d, %d' %('psRNAtarget', len(ps_mir), len(ps_target)))\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "with open('degradome_pstarget_overlaps.txt') as f:\n",
    "    for line in f:\n",
    "        line = line.rstrip()\n",
    "        lst = line.split('\\t')\n",
    "        \n",
    "        if lst[0] not in ovlp_mir:\n",
    "            ovlp_mir.append(lst[0])\n",
    "            \n",
    "        if lst[1] not in ovlp_target:\n",
    "            ovlp_target.append(lst[1])\n",
    "            \n",
    "            \n",
    "        if lst[0] not in overlaps.keys():\n",
    "            overlaps[lst[0]] = []\n",
    "           \n",
    "        xxx = lst[1].split('.')[0]+'.'+lst[1].split('.')[1]\n",
    "        if xxx not in overlaps[lst[0]]:\n",
    "            overlaps[lst[0]].append(xxx)\n",
    "\n",
    "        \n",
    "print ('number of miRNAs and targets in %s: %d, %d' %('overlaps', len(ovlp_mir), len(ovlp_target)))\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "with open('union_diffmir_target.txt') as f:\n",
    "    for line in f:\n",
    "        line = line.rstrip()\n",
    "        lst = line.split('\\t')\n",
    "        \n",
    "        if lst[0] not in ovlp_mir:\n",
    "            ovlp_mir.append(lst[0])\n",
    "            \n",
    "        if lst[1] not in ovlp_target:\n",
    "            ovlp_target.append(lst[1])\n",
    "            \n",
    "            \n",
    "        if lst[0] not in union.keys():\n",
    "            union[lst[0]] = []\n",
    "           \n",
    "        xxx = lst[1].split('.')[0]+'.'+lst[1].split('.')[1]\n",
    "        if xxx not in union[lst[0]]:\n",
    "            union[lst[0]].append(xxx)\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "fh1 = open('degradome_target.organized.2.txt', 'wt')\n",
    "fh2 = open('psRNATarget_diffmir_JGI.organized.2.txt', 'wt')\n",
    "fh3 = open('degradome_pstarget_overlaps.organized.2.txt', 'wt')\n",
    "fh4 = open('degradome_target.diffmir.organized.2.txt', 'wt')\n",
    "fh5 = open('union_diffmir_target.2.txt', 'wt')\n",
    "\n",
    "for ke, val in degradome.items():\n",
    "    fh1.write(ke+'\\t'+';'.join(val)+'\\n')\n",
    "    \n",
    "    if ke in diffMIR.keys():\n",
    "        fh4.write(ke+'\\t'+';'.join(val)+'\\n')\n",
    "    \n",
    "for ke, val in pstarget.items():\n",
    "    fh2.write(ke+'\\t'+';'.join(val)+'\\n')\n",
    "    \n",
    "for ke, val in overlaps.items():\n",
    "    fh3.write(ke+'\\t'+';'.join(val)+'\\n')\n",
    "    \n",
    "for ke, val in union.items():\n",
    "    fh5.write(ke+'\\t'+';'.join(val)+'\\n')\n",
    "\n",
    "    \n",
    "fh1.close()\n",
    "fh2.close()\n",
    "fh3.close()\n",
    "fh4.close()\n",
    "fh5.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 156,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "degradome= OrderedDict()\n",
    "pstarget = OrderedDict()\n",
    "overlaps = OrderedDict()\n",
    "degdiff = OrderedDict()\n",
    "\n",
    "with open('degradome_target.clean.txt') as f:\n",
    "    for line in f:\n",
    "    \n",
    "        line = line.rstrip()\n",
    "        lst = line.split('\\t')\n",
    "\n",
    "        if lst[1] not in degradome.keys():\n",
    "            degradome[lst[1]] = []\n",
    "            \n",
    "        degradome[lst[1]].append(lst[0])\n",
    "            \n",
    "            \n",
    "with open('psRNATarget_diffmir_JGI.txt') as f:\n",
    "    for line in f:\n",
    "        if line.startswith('#') or line.startswith('miRNA'):\n",
    "            continue\n",
    "            \n",
    "        line = line.rstrip()\n",
    "        lst = line.split('\\t')\n",
    "\n",
    "        if lst[1] not in pstarget.keys():\n",
    "            pstarget[lst[1]] = []\n",
    "            \n",
    "        pstarget[lst[1]].append(lst[0])\n",
    "            \n",
    "\n",
    "with open('degradome_pstarget_overlaps.txt') as f:\n",
    "    for line in f:\n",
    "        line = line.rstrip()\n",
    "        lst = line.split('\\t')\n",
    "            \n",
    "        if lst[1] not in overlaps.keys():\n",
    "            overlaps[lst[1]] = []\n",
    "            \n",
    "        overlaps[lst[1]].append(lst[0])\n",
    "    \n",
    "    \n",
    "with open('degradome_target.diffmir.clean.txt') as f:\n",
    "    for line in f:\n",
    "        line = line.rstrip()\n",
    "        lst = line.split('\\t')\n",
    "            \n",
    "        if lst[1] not in degdiff.keys():\n",
    "            degdiff[lst[1]] = []\n",
    "            \n",
    "        degdiff[lst[1]].append(lst[0])\n",
    "        \n",
    "\n",
    "\n",
    "fh1 = open('degradome_target.organized.gene.txt', 'wt')\n",
    "fh2 = open('psRNATarget_diffmir_JGI.organized.gene.txt', 'wt')\n",
    "fh3 = open('degradome_pstarget_overlaps.organized.gene.txt', 'wt')\n",
    "fh4 = open('degradome_target.diffmir.organized.gene.txt', 'wt')\n",
    "\n",
    "for ke, val in degradome.items():\n",
    "    fh1.write(ke+'\\t'+';'.join(val)+'\\n')\n",
    "    \n",
    "for ke, val in pstarget.items():\n",
    "    fh2.write(ke+'\\t'+';'.join(val)+'\\n')\n",
    "    \n",
    "for ke, val in overlaps.items():\n",
    "    fh3.write(ke+'\\t'+';'.join(val)+'\\n')\n",
    "    \n",
    "for ke, val in degdiff.items():\n",
    "    fh4.write(ke+'\\t'+';'.join(val)+'\\n') \n",
    "    \n",
    "    \n",
    "fh1.close()\n",
    "fh2.close()\n",
    "fh3.close()\n",
    "fh4.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "degradome= OrderedDict()\n",
    "pstarget = OrderedDict()\n",
    "overlaps = OrderedDict()\n",
    "degdiff = OrderedDict()\n",
    "union = OrderedDict()\n",
    "\n",
    "with open('degradome_target.clean.txt') as f:\n",
    "    for line in f:\n",
    "    \n",
    "        line = line.rstrip()\n",
    "        lst = line.split('\\t')\n",
    "        \n",
    "        xxx = lst[1].split('.')[0]+'.'+lst[1].split('.')[1]\n",
    "        \n",
    "        if xxx not in degradome.keys():\n",
    "            degradome[xxx] = []\n",
    "            \n",
    "        if lst[0] not in degradome[xxx]:\n",
    "            degradome[xxx].append(lst[0])\n",
    "            \n",
    "            \n",
    "with open('psRNATarget_diffmir_JGI.txt') as f:\n",
    "    for line in f:\n",
    "        if line.startswith('#') or line.startswith('miRNA'):\n",
    "            continue\n",
    "            \n",
    "        line = line.rstrip()\n",
    "        lst = line.split('\\t')\n",
    "\n",
    "        xxx = lst[1].split('.')[0]+'.'+lst[1].split('.')[1]\n",
    "        \n",
    "        if xxx not in pstarget.keys():\n",
    "            pstarget[xxx] = []\n",
    "        \n",
    "        if lst[0] not in pstarget[xxx]:    \n",
    "            pstarget[xxx].append(lst[0])\n",
    "            \n",
    "\n",
    "with open('degradome_pstarget_overlaps.txt') as f:\n",
    "    for line in f:\n",
    "        line = line.rstrip()\n",
    "        lst = line.split('\\t')\n",
    "            \n",
    "        xxx = lst[1].split('.')[0]+'.'+lst[1].split('.')[1]    \n",
    "            \n",
    "        if xxx not in overlaps.keys():\n",
    "            overlaps[xxx] = []\n",
    "            \n",
    "        if lst[0] not in overlaps[xxx]:\n",
    "            overlaps[xxx].append(lst[0])\n",
    "    \n",
    "    \n",
    "with open('degradome_target.diffmir.clean.txt') as f:\n",
    "    for line in f:\n",
    "        line = line.rstrip()\n",
    "        lst = line.split('\\t')\n",
    "            \n",
    "        xxx = lst[1].split('.')[0]+'.'+lst[1].split('.')[1]    \n",
    "        \n",
    "        if xxx not in degdiff.keys():\n",
    "            degdiff[xxx] = []\n",
    "            \n",
    "        if lst[0] not in degdiff[xxx]:\n",
    "            degdiff[xxx].append(lst[0])\n",
    "        \n",
    "    \n",
    "with open('union_diffmir_target.txt') as f:\n",
    "    for line in f:\n",
    "        line = line.rstrip()\n",
    "        lst = line.split('\\t')\n",
    "            \n",
    "        xxx = lst[1].split('.')[0]+'.'+lst[1].split('.')[1]    \n",
    "        \n",
    "        if xxx not in union.keys():\n",
    "            union[xxx] = []\n",
    "            \n",
    "        if lst[0] not in union[xxx]:\n",
    "            union[xxx].append(lst[0])\n",
    "        \n",
    "        \n",
    "        \n",
    "\n",
    "\n",
    "fh1 = open('degradome_target.organized.gene.2.txt', 'wt')\n",
    "fh2 = open('psRNATarget_diffmir_JGI.organized.gene.2.txt', 'wt')\n",
    "fh3 = open('degradome_pstarget_overlaps.organized.gene.2.txt', 'wt')\n",
    "fh4 = open('degradome_target.diffmir.organized.gene.2.txt', 'wt')\n",
    "fh5 = open('union_diffmir_target.organized.gene.2.txt', 'wt')\n",
    "\n",
    "for ke, val in degradome.items():\n",
    "    fh1.write(ke+'\\t'+';'.join(val)+'\\n')\n",
    "    \n",
    "for ke, val in pstarget.items():\n",
    "    fh2.write(ke+'\\t'+';'.join(val)+'\\n')\n",
    "    \n",
    "for ke, val in overlaps.items():\n",
    "    fh3.write(ke+'\\t'+';'.join(val)+'\\n')\n",
    "    \n",
    "for ke, val in degdiff.items():\n",
    "    fh4.write(ke+'\\t'+';'.join(val)+'\\n') \n",
    "    \n",
    "for ke, val in union.items():\n",
    "    fh5.write(ke+'\\t'+';'.join(val)+'\\n')   \n",
    "\n",
    "    \n",
    "    \n",
    "fh1.close()\n",
    "fh2.close()\n",
    "fh3.close()\n",
    "fh4.close()\n",
    "fh5.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Glyma.09G068400\n",
      "Glyma.13G107900\n",
      "Glyma.15G027400\n",
      "Glyma.15G173300\n",
      "Glyma.17G051400\n",
      "Glyma.18G071000\n",
      "Glyma.01G188400\n",
      "Glyma.02G087400\n",
      "Glyma.03G177500\n",
      "Glyma.05G091200\n",
      "Glyma.10G116600\n",
      "Glyma.11G053800\n",
      "Glyma.12G073300\n",
      "Glyma.13G329700\n",
      "Glyma.15G044400\n",
      "Glyma.17G170300\n",
      "Glyma.19G178200\n",
      "Glyma.U022100\n",
      "Glyma.14G156400\n",
      "Glyma.05G147900\n",
      "Glyma.08G104800\n",
      "Glyma.18G005400\n",
      "Glyma.13G087100\n",
      "Glyma.18G168900\n",
      "Glyma.03G077900\n",
      "Glyma.06G284300\n",
      "Glyma.08G359100\n",
      "Glyma.12G121700\n",
      "Glyma.18G065100\n",
      "Glyma.09G171200\n",
      "Glyma.09G174800\n",
      "Glyma.09G176000\n",
      "Glyma.09G176800\n",
      "Glyma.09G178300\n"
     ]
    }
   ],
   "source": [
    "gene = []\n",
    "\n",
    "\n",
    "fh = open('summary_important_transcript.txt', 'wt')\n",
    "with open('degradome_pstarget_overlaps.organized.gene.txt') as f:\n",
    "    for line in f:\n",
    "        line = line.rstrip()\n",
    "        lst = line.split('\\t')\n",
    "        \n",
    "        trpt = lst[0].split('.')\n",
    "        \n",
    "        if trpt[0]+'.'+trpt[1] not in gene:\n",
    "            gene.append(trpt[0]+'.'+trpt[1])\n",
    "        \n",
    "            print (trpt[0]+'.'+trpt[1])\n",
    "        \n",
    "with open('soybase_genome_annotation_02-16-2017.txt') as f:\n",
    "    for line in f:\n",
    "        line = line.rstrip()\n",
    "        lst = line.split('\\t')\n",
    "        if lst[0] in gene:\n",
    "            fh.write(line+'\\n')\n",
    "\n",
    "fh.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "gene = []\n",
    "\n",
    "fh = open('summary.degradome_target.diff.anno.txt', 'wt')\n",
    "with open('degradome_target.organized.gene.txt') as f:\n",
    "    for line in f:\n",
    "        line = line.rstrip()\n",
    "        lst = line.split('\\t')\n",
    "        \n",
    "        trpt = lst[0].split('.')\n",
    "        \n",
    "        if trpt[0]+'.'+trpt[1] not in gene:\n",
    "            gene.append(trpt[0]+'.'+trpt[1])\n",
    "\n",
    "        \n",
    "with open('genes.diff_exp_result.anno') as f:\n",
    "    for line in f:\n",
    "        line = line.rstrip()\n",
    "        lst = line.split('\\t')\n",
    "        \n",
    "        if line.startswith('gene_id'):\n",
    "            fh.write(line+'\\n')\n",
    "        if lst[0] in gene:\n",
    "            fh.write(line+'\\n')\n",
    "\n",
    "fh.close()\n",
    "\n",
    "\n",
    "fh = open('degradome_target.diff.id.txt', 'wt')\n",
    "with open('geneID_NCBI.clean.txt') as f:\n",
    "    for line in f:\n",
    "        line = line.rstrip()\n",
    "        lst = line.split('\\t')\n",
    "        \n",
    "        if line.startswith('gene_name'):\n",
    "            fh.write(line+'\\n')\n",
    "        if lst[0] in gene:\n",
    "            fh.write(line+'\\n')\n",
    "\n",
    "fh.close()\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "gene = []\n",
    "\n",
    "fh = open('summary.psRNATarget_diffmir_JGI.diff.anno.txt', 'wt')\n",
    "with open('psRNATarget_diffmir_JGI.organized.gene.txt') as f:\n",
    "    for line in f:\n",
    "        line = line.rstrip()\n",
    "        lst = line.split('\\t')\n",
    "        \n",
    "        trpt = lst[0].split('.')\n",
    "        \n",
    "        if trpt[0]+'.'+trpt[1] not in gene:\n",
    "            gene.append(trpt[0]+'.'+trpt[1])\n",
    "\n",
    "        \n",
    "with open('genes.diff_exp_result.anno') as f:\n",
    "    for line in f:\n",
    "        line = line.rstrip()\n",
    "        lst = line.split('\\t')\n",
    "        \n",
    "        if line.startswith('gene_id'):\n",
    "            fh.write(line+'\\n')\n",
    "        if lst[0] in gene:\n",
    "            fh.write(line+'\\n')\n",
    "\n",
    "fh.close()\n",
    "\n",
    "\n",
    "\n",
    "fh = open('psRNATarget_diffmir_JGI.diff.id.txt', 'wt')\n",
    "with open('geneID_NCBI.clean.txt') as f:\n",
    "    for line in f:\n",
    "        line = line.rstrip()\n",
    "        lst = line.split('\\t')\n",
    "        \n",
    "        if line.startswith('gene_name'):\n",
    "            fh.write(line+'\\n')\n",
    "        if lst[0] in gene:\n",
    "            fh.write(line+'\\n')\n",
    "\n",
    "fh.close()\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "gene = []\n",
    "\n",
    "fh = open('summary.degradome_pstarget_overlaps.diff.anno.txt', 'wt')\n",
    "with open('degradome_pstarget_overlaps.organized.gene.txt') as f:\n",
    "    for line in f:\n",
    "        line = line.rstrip()\n",
    "        lst = line.split('\\t')\n",
    "        \n",
    "        trpt = lst[0].split('.')\n",
    "        \n",
    "        if trpt[0]+'.'+trpt[1] not in gene:\n",
    "            gene.append(trpt[0]+'.'+trpt[1])\n",
    "        \n",
    "        \n",
    "with open('genes.diff_exp_result.anno') as f:\n",
    "    for line in f:\n",
    "        line = line.rstrip()\n",
    "        lst = line.split('\\t')\n",
    "        \n",
    "        if line.startswith('gene_id'):\n",
    "            fh.write(line+'\\n')\n",
    "        if lst[0] in gene:\n",
    "            fh.write(line+'\\n')\n",
    "\n",
    "fh.close()\n",
    "\n",
    "\n",
    "fh = open('degradome_pstarget_overlaps.id.txt', 'wt')\n",
    "with open('geneID_NCBI.clean.txt') as f:\n",
    "    for line in f:\n",
    "        line = line.rstrip()\n",
    "        lst = line.split('\\t')\n",
    "        \n",
    "        if line.startswith('gene_name'):\n",
    "            fh.write(line+'\\n')\n",
    "        if lst[0] in gene:\n",
    "            fh.write(line+'\\n')\n",
    "\n",
    "fh.close()\n",
    "\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "gene = []\n",
    "\n",
    "fh = open('summary.degradome_target.soybase.anno.txt', 'wt')\n",
    "with open('degradome_target.organized.gene.txt') as f:\n",
    "    for line in f:\n",
    "        line = line.rstrip()\n",
    "        lst = line.split('\\t')\n",
    "        \n",
    "        trpt = lst[0].split('.')\n",
    "        \n",
    "        if trpt[0]+'.'+trpt[1] not in gene:\n",
    "            gene.append(trpt[0]+'.'+trpt[1])\n",
    "        \n",
    "            print (trpt[0]+'.'+trpt[1])\n",
    "        \n",
    "with open('soybase_genome_annotation_02-16-2017.txt') as f:\n",
    "    for line in f:\n",
    "        line = line.rstrip()\n",
    "        lst = line.split('\\t')\n",
    "        \n",
    "        if line.startswith('Gmax'):\n",
    "            fh.write(line+'\\n')\n",
    "        if lst[0] in gene:\n",
    "            fh.write(line+'\\n')\n",
    "\n",
    "fh.close()\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "gene = []\n",
    "\n",
    "fh = open('summary.psRNATarget_diffmir_JGI.soybase.anno.txt', 'wt')\n",
    "with open('psRNATarget_diffmir_JGI.organized.gene.txt') as f:\n",
    "    for line in f:\n",
    "        line = line.rstrip()\n",
    "        lst = line.split('\\t')\n",
    "        \n",
    "        trpt = lst[0].split('.')\n",
    "        \n",
    "        if trpt[0]+'.'+trpt[1] not in gene:\n",
    "            gene.append(trpt[0]+'.'+trpt[1])\n",
    "        \n",
    "            print (trpt[0]+'.'+trpt[1])\n",
    "        \n",
    "with open('soybase_genome_annotation_02-16-2017.txt') as f:\n",
    "    for line in f:\n",
    "        line = line.rstrip()\n",
    "        lst = line.split('\\t')\n",
    "        \n",
    "        if line.startswith('Gmax'):\n",
    "            fh.write(line+'\\n')\n",
    "        if lst[0] in gene:\n",
    "            fh.write(line+'\\n')\n",
    "\n",
    "fh.close()\n",
    "\n",
    "\n",
    "\n",
    "gene = []\n",
    "\n",
    "fh = open('summary.degradome_pstarget_overlaps.soybase.anno.txt', 'wt')\n",
    "with open('degradome_pstarget_overlaps.organized.gene.txt') as f:\n",
    "    for line in f:\n",
    "        line = line.rstrip()\n",
    "        lst = line.split('\\t')\n",
    "        \n",
    "        trpt = lst[0].split('.')\n",
    "        \n",
    "        if trpt[0]+'.'+trpt[1] not in gene:\n",
    "            gene.append(trpt[0]+'.'+trpt[1])\n",
    "        \n",
    "        \n",
    "with open('soybase_genome_annotation_02-16-2017.txt') as f:\n",
    "    for line in f:\n",
    "        line = line.rstrip()\n",
    "        lst = line.split('\\t')\n",
    "        \n",
    "        if line.startswith('Gmax'):\n",
    "            fh.write(line+'\\n')\n",
    "        if lst[0] in gene:\n",
    "            fh.write(line+'\\n')\n",
    "\n",
    "fh.close()\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 163,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "gene = []\n",
    "\n",
    "with open('union_target.organized.gene.txt') as f:\n",
    "    for line in f:\n",
    "        line = line.rstrip()\n",
    "        lst = line.split('\\t')\n",
    "        \n",
    "        trpt = lst[0].split('.')\n",
    "        \n",
    "        if trpt[0]+'.'+trpt[1] not in gene:\n",
    "            gene.append(trpt[0]+'.'+trpt[1])\n",
    "\n",
    "\n",
    "fh = open('union_target.organized.gene.id.txt', 'wt')\n",
    "with open('geneID_NCBI.clean.txt') as f:\n",
    "    for line in f:\n",
    "        line = line.rstrip()\n",
    "        lst = line.split('\\t')\n",
    "        \n",
    "        if line.startswith('gene_name'):\n",
    "            fh.write(line+'\\n')\n",
    "        if lst[0] in gene:\n",
    "            fh.write(line+'\\n')\n",
    "\n",
    "fh.close()\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "hsaKEGG = OrderedDict()\n",
    "keggEnrichment = OrderedDict()\n",
    "with open('gmx00001.keg') as f:\n",
    "    for line in f:\n",
    "        line = line.rstrip()\n",
    "        if line.startswith('A'):\n",
    "            mch = re.search('A<b>(.+)</b>',line)\n",
    "            className = mch.group(1)\n",
    "            hsaKEGG[className] = OrderedDict()\n",
    "                \n",
    "        elif line.startswith('B'):\n",
    "            if line == 'B':\n",
    "                continue\n",
    "            else:\n",
    "                mch = re.search('<b>(.+)</b>', line)\n",
    "                subClass = mch.group(1)\n",
    "                hsaKEGG[className][subClass] = OrderedDict()\n",
    "                \n",
    "        elif line.startswith('C'):\n",
    "            mch = re.search('(\\d+)\\s(.+)', line)\n",
    "            pathID = 'gmx'+mch.group(1)\n",
    "            pathName = re.sub('\\s\\[.+\\]', '', mch.group(2))\n",
    "            pathway = pathID + ':' + pathName\n",
    "            \n",
    "            hsaKEGG[className][subClass][pathway] = [[],[]]\n",
    "            \n",
    "            #keggEnrichment[pathway] = []\n",
    "        \n",
    "        elif line.startswith('D'):\n",
    "            lst = line.split(';')\n",
    "            #gene = lst[0].split(' ')[-1]\n",
    "            geneID = re.split(r'\\s+',lst[0])[1]\n",
    "            print (geneID)\n",
    "            #hsaKEGG[className][subClass][pathway][0].append(gene)\n",
    "            hsaKEGG[className][subClass][pathway][1].append(geneID)\n",
    "            \n",
    "            #keggEnrichment[pathway].append(gene)\n",
    "            \n",
    "fh = open('gmx00001.cleaned.keg', 'wt')\n",
    "for ke, val in hsaKEGG.items():\n",
    "    for subk, subv in val.items():\n",
    "        for ptwy, geneList in subv.items():\n",
    "            #genes = ';'.join(geneList[0])\n",
    "            geneIDs = ';'.join(geneList[1])\n",
    "            fh.write('\\t'.join([ke, subk, ptwy,geneIDs])+'\\n')\n",
    "            \n",
    "            \n",
    "fh.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "fh = open('geneID_NCBI.all.clean.txt', 'wt')\n",
    "with open('geneID_NCBI.all.txt') as f:\n",
    "    for line in f:\n",
    "        line = line.rstrip()\n",
    "        lst = line.split('\\t')\n",
    "        gene = lst[6].split(',')\n",
    "        if line.startswith('tax_id'):\n",
    "            fh.write('\\t'.join(['gene_name', lst[2], lst[5], lst[7], lst[8]])+'\\n')\n",
    "        \n",
    "        \n",
    "        elif gene[0].startswith('GLYMA'):\n",
    "            gene[0] = gene[0].replace('LYMA_', 'lyma.')\n",
    "            fh.write('\\t'.join([gene[0], lst[2], lst[5], lst[7], lst[8]])+'\\n')\n",
    "                     \n",
    "                     \n",
    "fh.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 161,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "##########################\n",
    "###### GO\n",
    "\n",
    "goBasic = OrderedDict()\n",
    "with open('go-basic-02142017.obo.txt') as f:\n",
    "    for line in f:\n",
    "        line = line.rstrip()\n",
    "        lst = line.split('\\t')\n",
    "        goBasic[lst[1]] = lst[2]\n",
    "        \n",
    "goBasic['NA'] = 'NA'\n",
    "\n",
    "\n",
    "goBP = OrderedDict()\n",
    "goMF = OrderedDict()\n",
    "goCC = OrderedDict()\n",
    "\n",
    "pfam = OrderedDict()\n",
    "panther = OrderedDict()\n",
    "kog = OrderedDict()\n",
    "\n",
    "\n",
    "def smry_anno_go(lst,goBP, j=11):\n",
    "    bpID = lst[j].split(' ')\n",
    "    bpSmry = [k+'\\t'+goBasic[k] if k in goBasic.keys() else k+'\\tNA' for k in bpID]\n",
    "        \n",
    "    for bp in bpSmry:\n",
    "        if bp not in goBP.keys():\n",
    "            goBP[bp] = []\n",
    "        goBP[bp].append(lst[0])\n",
    "\n",
    "\n",
    "def smry_anno(lst,pfam, j=17):\n",
    "    bpID = lst[j].split(' ')\n",
    "    bpTerm = lst[j+1].split('; ')\n",
    "    bpSmry = [bpID[i]+'\\t'+bpTerm[i] for i in range(len(bpID))]\n",
    "        \n",
    "    for bp in bpSmry:\n",
    "        if bp not in pfam.keys():\n",
    "            pfam[bp] = []\n",
    "        pfam[bp].append(lst[0])\n",
    "\n",
    "\n",
    "with open('summary.degradome_target_diffmir.soybase.anno.txt') as f:\n",
    "    for line in f:\n",
    "        line = line.rstrip()\n",
    "        line = line.replace('\"', '')\n",
    "        lst = line.split('\\t')\n",
    "       \n",
    "        if line.startswith('Gmax'):\n",
    "            continue\n",
    "            \n",
    "        smry_anno_go(lst,goBP, j=11)\n",
    "        smry_anno_go(lst,goMF, j=13)\n",
    "        smry_anno_go(lst,goCC, j=15)\n",
    "        smry_anno(lst,pfam, j=17)\n",
    "        smry_anno(lst,panther, j=19)\n",
    "        smry_anno(lst,kog, j=21)\n",
    "        \n",
    "        \n",
    "        goBPsort = [(ke, goBP[ke]) for ke in sorted(goBP, key=lambda k: len(goBP[k]), reverse=True)]\n",
    "        goMFsort = [(ke, goMF[ke]) for ke in sorted(goMF, key=lambda k: len(goMF[k]), reverse=True)]\n",
    "        goCCsort = [(ke, goCC[ke]) for ke in sorted(goCC, key=lambda k: len(goCC[k]), reverse=True)]\n",
    "        pfamsort = [(ke, pfam[ke]) for ke in sorted(pfam, key=lambda k: len(pfam[k]), reverse=True)]\n",
    "        panthersort = [(ke, panther[ke]) for ke in sorted(panther, key=lambda k: len(panther[k]), reverse=True)]\n",
    "        kogsort = [(ke, kog[ke]) for ke in sorted(kog, key=lambda k: len(kog[k]), reverse=True)]\n",
    "        \n",
    "        \n",
    "fh = open('deg_diffmir_target_soybase_anno.goBP.txt', 'wt')\n",
    "for ke, val in goBPsort:\n",
    "    fh.write(ke+'\\t'+str(len(val))+'\\t'+';'.join(val)+'\\n')\n",
    "    \n",
    "fh.close()\n",
    "\n",
    "\n",
    "fh = open('deg_diffmir_target_soybase_anno.goMF.txt', 'wt')\n",
    "for ke, val in goMFsort:\n",
    "    fh.write(ke+'\\t'+str(len(val))+'\\t'+';'.join(val)+'\\n')\n",
    "    \n",
    "fh.close()\n",
    "\n",
    "\n",
    "fh = open('deg_diffmir_target_soybase_anno.goCC.txt', 'wt')\n",
    "for ke, val in goCCsort:\n",
    "    fh.write(ke+'\\t'+str(len(val))+'\\t'+';'.join(val)+'\\n')\n",
    "    \n",
    "fh.close()\n",
    "\n",
    "\n",
    "fh = open('deg_diffmir_target_soybase_anno.PFAM.txt', 'wt')\n",
    "for ke, val in pfamsort:\n",
    "    fh.write(ke+'\\t'+str(len(val))+'\\t'+';'.join(val)+'\\n')\n",
    "    \n",
    "fh.close()\n",
    "\n",
    "\n",
    "fh = open('deg_diffmir_target_soybase_anno.Panther.txt', 'wt')\n",
    "for ke, val in panthersort:\n",
    "    fh.write(ke+'\\t'+str(len(val))+'\\t'+';'.join(val)+'\\n')\n",
    "    \n",
    "fh.close()\n",
    "\n",
    "\n",
    "fh = open('deg_diffmir_target_soybase_anno.KOG.txt', 'wt')\n",
    "for ke, val in kogsort:\n",
    "    fh.write(ke+'\\t'+str(len(val))+'\\t'+';'.join(val)+'\\n')\n",
    "    \n",
    "fh.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 127,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "#############################\n",
    "######## kegg\n",
    "\n",
    "geneList = OrderedDict()\n",
    "\n",
    "fh = open('deg_target_soybase_anno.kegg.txt', 'wt')\n",
    "with open('degradome_target.diff.id.txt') as f:\n",
    "    for line in f:\n",
    "        line = line.rstrip()\n",
    "        lst = line.split('\\t')\n",
    "        geneList[lst[1]] = lst[0]\n",
    "        \n",
    "\n",
    "kegg = OrderedDict()\n",
    "\n",
    "with open('gmx00001.cleaned.keg') as f:\n",
    "    for line in f:\n",
    "        line = line.rstrip()\n",
    "        lst = line.split('\\t')\n",
    "        geneKEGG = lst[-1].split(';')\n",
    "        \n",
    "        ovlp = [geneList[gene] for gene in geneKEGG if gene in geneList.keys()]\n",
    "        if ovlp == []:\n",
    "            continue\n",
    "        else:\n",
    "            kegg['\\t'.join(lst[0:3])] = ovlp\n",
    "\n",
    "\n",
    "keggsort = [(ke, kegg[ke]) for ke in sorted(kegg, key=lambda k: len(kegg[k]), reverse=True)]\n",
    "for ke, val in keggsort:\n",
    "    fh.write('\\t'.join([ke,str(len(val)), ';'.join(val)])+'\\n')\n",
    "    \n",
    "    \n",
    "fh.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 153,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "##################################\n",
    "######## union\n",
    "\n",
    "unionTarget = OrderedDict()\n",
    "with open('degradome_target.diffmir.clean.txt') as f:\n",
    "    for line in f:\n",
    "        line = line.rstrip()\n",
    "        unionTarget[line] = 'degradome'\n",
    "        \n",
    "\n",
    "with open('psRNATarget_diffmir_JGI.clean.txt') as f:\n",
    "    for line in f:\n",
    "        line = line.rstrip()\n",
    "        if line in unionTarget.keys():\n",
    "            unionTarget[line] = 'both'\n",
    "        else:\n",
    "            unionTarget[line] = 'psRNAtarget'\n",
    "            \n",
    "            \n",
    "unionTargetsort = sorted(unionTarget.items(), key = itemgetter(0))\n",
    "\n",
    "\n",
    "fh = open('union_diffmir_target.txt', 'wt')\n",
    "for ke, val in unionTargetsort:\n",
    "    fh.write(ke+'\\t'+val+'\\n')\n",
    "    \n",
    "fh.close()\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 154,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "number of miRNAs and targets in degradome: 76, 381\n"
     ]
    }
   ],
   "source": [
    "union_mir = []\n",
    "union_target = []\n",
    "\n",
    "union = OrderedDict()\n",
    "\n",
    "\n",
    "with open('union_diffmir_target.txt') as f:\n",
    "    for line in f:\n",
    "    \n",
    "        line = line.rstrip()\n",
    "        lst = line.split('\\t')\n",
    "        \n",
    "        if lst[0] not in union_mir:\n",
    "            union_mir.append(lst[0])\n",
    "            \n",
    "        if lst[1] not in union_target:\n",
    "            union_target.append(lst[1])\n",
    "            \n",
    "        if lst[0] not in union.keys():\n",
    "            union[lst[0]] = []\n",
    "            \n",
    "        union[lst[0]].append(lst[1])\n",
    "            \n",
    "            \n",
    "print ('number of miRNAs and targets in %s: %d, %d' %('degradome', len(union_mir), len(union_target)))\n",
    "\n",
    "\n",
    "fh1 = open('union_target.organized.txt', 'wt')\n",
    "\n",
    "for ke, val in union.items():\n",
    "    fh1.write(ke+'\\t'+';'.join(val)+'\\n')\n",
    "    \n",
    "\n",
    "fh1.close()\n",
    "\n",
    "\n",
    "\n",
    "union = OrderedDict()\n",
    "\n",
    "with open('union_diffmir_target.txt') as f:\n",
    "    for line in f:\n",
    "    \n",
    "        line = line.rstrip()\n",
    "        lst = line.split('\\t')\n",
    "\n",
    "        if lst[1] not in union.keys():\n",
    "            union[lst[1]] = []\n",
    "            \n",
    "        union[lst[1]].append(lst[0])\n",
    "            \n",
    "fh1 = open('union_target.organized.gene.txt', 'wt')\n",
    "\n",
    "for ke, val in union.items():\n",
    "    fh1.write(ke+'\\t'+';'.join(val)+'\\n')\n",
    "\n",
    "fh1.close()\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "unionTarget = OrderedDict()\n",
    "with open('degradome_target.diffmir.clean.txt') as f:\n",
    "    for line in f:\n",
    "        line = line.rstrip()\n",
    "        unionTarget[line] = 'degradome'\n",
    "        \n",
    "\n",
    "with open('psRNATarget_diffmir_JGI.clean.txt') as f:\n",
    "    for line in f:\n",
    "        line = line.rstrip()\n",
    "        if line in unionTarget.keys():\n",
    "            unionTarget[line] = 'both'\n",
    "        else:\n",
    "            unionTarget[line] = 'psRNAtarget'\n",
    "            \n",
    "            \n",
    "unionTargetsort = sorted(unionTarget.items(), key = itemgetter(0))\n",
    "\n",
    "\n",
    "fh = open('union_diffmir_target.txt', 'wt')\n",
    "for ke, val in unionTargetsort:\n",
    "    fh.write(ke+'\\t'+val+'\\n')\n",
    "    \n",
    "fh.close()\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 158,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "gene = []\n",
    "\n",
    "fh = open('summary.degradome_target_diffmir.diff.anno.txt', 'wt')\n",
    "with open('degradome_target.diffmir.organized.gene.txt') as f:\n",
    "    for line in f:\n",
    "        line = line.rstrip()\n",
    "        lst = line.split('\\t')\n",
    "        \n",
    "        trpt = lst[0].split('.')\n",
    "        \n",
    "        if trpt[0]+'.'+trpt[1] not in gene:\n",
    "            gene.append(trpt[0]+'.'+trpt[1])\n",
    "\n",
    "        \n",
    "with open('genes.diff_exp_result.anno') as f:\n",
    "    for line in f:\n",
    "        line = line.rstrip()\n",
    "        lst = line.split('\\t')\n",
    "        \n",
    "        if line.startswith('gene_id'):\n",
    "            fh.write(line+'\\n')\n",
    "        if lst[0] in gene:\n",
    "            fh.write(line+'\\n')\n",
    "\n",
    "fh.close()\n",
    "\n",
    "\n",
    "\n",
    "gene = []\n",
    "\n",
    "fh = open('summary.degradome_pstarget_union.diff.anno.txt', 'wt')\n",
    "with open('union_target.organized.gene.txt') as f:\n",
    "    for line in f:\n",
    "        line = line.rstrip()\n",
    "        lst = line.split('\\t')\n",
    "        \n",
    "        trpt = lst[0].split('.')\n",
    "        \n",
    "        if trpt[0]+'.'+trpt[1] not in gene:\n",
    "            gene.append(trpt[0]+'.'+trpt[1])\n",
    "\n",
    "        \n",
    "with open('genes.diff_exp_result.anno') as f:\n",
    "    for line in f:\n",
    "        line = line.rstrip()\n",
    "        lst = line.split('\\t')\n",
    "        \n",
    "        if line.startswith('gene_id'):\n",
    "            fh.write(line+'\\n')\n",
    "        if lst[0] in gene:\n",
    "            fh.write(line+'\\n')\n",
    "\n",
    "fh.close()\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 159,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "gene = []\n",
    "\n",
    "fh = open('summary.degradome_target_diffmir.soybase.anno.txt', 'wt')\n",
    "with open('degradome_target.diffmir.organized.gene.txt') as f:\n",
    "    for line in f:\n",
    "        line = line.rstrip()\n",
    "        lst = line.split('\\t')\n",
    "        \n",
    "        trpt = lst[0].split('.')\n",
    "        \n",
    "        if trpt[0]+'.'+trpt[1] not in gene:\n",
    "            gene.append(trpt[0]+'.'+trpt[1])\n",
    "        \n",
    "        \n",
    "with open('soybase_genome_annotation_02-16-2017.txt') as f:\n",
    "    for line in f:\n",
    "        line = line.rstrip()\n",
    "        lst = line.split('\\t')\n",
    "        \n",
    "        if line.startswith('Gmax'):\n",
    "            fh.write(line+'\\n')\n",
    "        if lst[0] in gene:\n",
    "            fh.write(line+'\\n')\n",
    "\n",
    "fh.close()\n",
    "\n",
    "\n",
    "gene = []\n",
    "\n",
    "fh = open('summary.degradome_pstarget_union.soybase.anno.txt', 'wt')\n",
    "with open('union_target.organized.gene.txt') as f:\n",
    "    for line in f:\n",
    "        line = line.rstrip()\n",
    "        lst = line.split('\\t')\n",
    "        \n",
    "        trpt = lst[0].split('.')\n",
    "        \n",
    "        if trpt[0]+'.'+trpt[1] not in gene:\n",
    "            gene.append(trpt[0]+'.'+trpt[1])\n",
    "        \n",
    "        \n",
    "with open('soybase_genome_annotation_02-16-2017.txt') as f:\n",
    "    for line in f:\n",
    "        line = line.rstrip()\n",
    "        lst = line.split('\\t')\n",
    "        \n",
    "        if line.startswith('Gmax'):\n",
    "            fh.write(line+'\\n')\n",
    "        if lst[0] in gene:\n",
    "            fh.write(line+'\\n')\n",
    "\n",
    "fh.close()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 165,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#############################\n",
    "######## kegg\n",
    "\n",
    "geneList = OrderedDict()\n",
    "\n",
    "fh = open('union_target_soybase_anno.kegg.txt', 'wt')\n",
    "with open('union_target.organized.gene.id.txt') as f:\n",
    "    for line in f:\n",
    "        line = line.rstrip()\n",
    "        lst = line.split('\\t')\n",
    "        geneList[lst[1]] = lst[0]\n",
    "        \n",
    "\n",
    "kegg = OrderedDict()\n",
    "\n",
    "with open('gmx00001.cleaned.keg') as f:\n",
    "    for line in f:\n",
    "        line = line.rstrip()\n",
    "        lst = line.split('\\t')\n",
    "        geneKEGG = lst[-1].split(';')\n",
    "        \n",
    "        ovlp = [geneList[gene] for gene in geneKEGG if gene in geneList.keys()]\n",
    "        if ovlp == []:\n",
    "            continue\n",
    "        else:\n",
    "            kegg['\\t'.join(lst[0:3])] = ovlp\n",
    "\n",
    "\n",
    "keggsort = [(ke, kegg[ke]) for ke in sorted(kegg, key=lambda k: len(kegg[k]), reverse=True)]\n",
    "for ke, val in keggsort:\n",
    "    fh.write('\\t'.join([ke,str(len(val)), ';'.join(val)])+'\\n')\n",
    "    \n",
    "    \n",
    "fh.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 137,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "791\n"
     ]
    }
   ],
   "source": [
    "pstarget = []\n",
    "\n",
    "fh = open('psRNATarget_diffmir_JGI.clean.txt', 'wt')\n",
    "with open('psRNATarget_diffmir_JGI.txt') as f:\n",
    "    for line in f:\n",
    "        if line.startswith('#') or line.startswith('miRNA'):\n",
    "            continue\n",
    "        line = line.rstrip()\n",
    "        lst = line.split('\\t')\n",
    "       \n",
    "        if lst[0]+'\\t'+lst[1] not in pstarget:\n",
    "            pstarget.append(lst[0]+'\\t'+lst[1])\n",
    "            fh.write(lst[0]+'\\t'+lst[1]+'\\n')\n",
    "            \n",
    "fh.close()\n",
    "print (len(pstarget))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 141,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "251\n"
     ]
    }
   ],
   "source": [
    "check = OrderedDict()\n",
    "\n",
    "with open('degradome_pstarget_overlaps.txt') as f:\n",
    "    for line in f:\n",
    "        line = line.rstrip()\n",
    "        if line not in check.keys():\n",
    "            check[line] = 1\n",
    "            \n",
    "            \n",
    "print (len(check))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 169,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2969\n",
      "3373\n",
      "0.8802253187073822\n"
     ]
    }
   ],
   "source": [
    "deg = OrderedDict()\n",
    "with open('degradome_target.clean.txt') as f:\n",
    "    for line in f:\n",
    "        line = line.rstrip()\n",
    "        deg[line] = 1\n",
    "\n",
    "\n",
    "i=0       \n",
    "with open('psRNATarget_allmir_JGI.txt') as f:\n",
    "    for line in f:\n",
    "        line = line.rstrip()\n",
    "        lst = line.split('\\t')\n",
    "        if line.startswith('#'):\n",
    "            continue\n",
    "        gene = lst[0]+'\\t'+lst[1]\n",
    "        if gene in deg.keys():\n",
    "            i += 1\n",
    "            \n",
    "print (i)\n",
    "print (len(deg))\n",
    "print (i/len(deg))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 170,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "interestGene = []\n",
    "with open('union_target.organized.txt') as f:\n",
    "    for line in f:\n",
    "        line = line.rstrip()\n",
    "        lst = line.split('\\t')\n",
    "        interestGene.append(lst[0])\n",
    "        \n",
    "        \n",
    "with open('union_target.organized.gene.txt') as f:\n",
    "    for line in f:\n",
    "        line = line.rstrip()\n",
    "        lst = line.split('\\t')\n",
    "        interestGene.append(lst[0])\n",
    "\n",
    "        \n",
    "        \n",
    "fh = open('expresion_data_for_heatmap.txt', 'wt')\n",
    "\n",
    "fh.write('\\t'+'NJCMS1A'+'\\t'+'NJCMS1B'+'\\n')\n",
    "with open('all_miRNA_expression.txt') as f:\n",
    "    for line in f:\n",
    "        line = line.rstrip()\n",
    "        lst = line.split('\\t')\n",
    "        if lst[0] in interestGene:\n",
    "            fh.write(line+'\\n')\n",
    "            \n",
    "with open('all_mRNA_expression.txt') as f:\n",
    "    for line in f:\n",
    "        line = line.rstrip()\n",
    "        lst = line.split('\\t')\n",
    "        if lst[0] in interestGene:\n",
    "            fh.write(line+'\\n')\n",
    "            \n",
    "fh.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "\n",
    "mir = ['gma-miR166m','gma-miR169d','gma-miR169e','gma-miR169s',\n",
    "       'gma-miR172g','gma-miR172i-3p','gma-miR172k','gma-miR172l','gma-miR2119',\n",
    "       'gma-miR396b-3p','gma-miR396d','gma-miR396e-3p','gma-miR396h-3p','gma-miR396j',\n",
    "       'gma-miR396k-3p','gma-miR397a','gma-miR397b-3p','gma-miR397b-5p',\n",
    "       'gma-miR5674a','gma-miR5674b','N-gma-miR164r-5p']\n",
    "\n",
    "mirTarget = OrderedDict()\n",
    "for m in mir:\n",
    "    mirTarget[m] = []\n",
    "\n",
    "\n",
    "fh = open('union_diffmir_target.n.txt','wt')\n",
    "with open('union_diffmir_target.txt') as f:\n",
    "    for line in f:\n",
    "        line = line.rstrip()\n",
    "        lst = line.split('\\t')\n",
    "        \n",
    "        if lst[0] not in mir:\n",
    "            continue\n",
    "        if lst[1].split('.')[0]+'.'+lst[1].split('.')[1] not in mirTarget[lst[0]]:\n",
    "            mirTarget[lst[0]].append(lst[1].split('.')[0]+'.'+lst[1].split('.')[1])\n",
    "            fh.write(lst[0]+'\\t'+lst[1].split('.')[0]+'.'+lst[1].split('.')[1]+'\\n')\n",
    "            \n",
    "            \n",
    "fh.close()\n",
    "        "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "21\n",
      "106\n"
     ]
    }
   ],
   "source": [
    "mirC = []\n",
    "targetC = []\n",
    "\n",
    "fh = open('CYTOSCAPE.node','wt')\n",
    "with open('union_diffmir_target.n.txt') as f:\n",
    "    for line in f:\n",
    "        line = line.rstrip()\n",
    "        lst = line.split('\\t')\n",
    "        \n",
    "        if lst[0] not in mirC:\n",
    "            fh.write(lst[0]+'\\t'+'mir'+'\\n')\n",
    "            mirC.append(lst[0])\n",
    "            \n",
    "        if lst[1] not in targetC:\n",
    "            fh.write(lst[1]+'\\t'+'target'+'\\n')\n",
    "            targetC.append(lst[1])\n",
    "            \n",
    "            \n",
    "print (len(mirC))\n",
    "print (len(targetC))\n",
    "\n",
    "fh.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "mirTarget = OrderedDict()\n",
    "\n",
    "fh = open('cytoscape.all.edge','wt')\n",
    "\n",
    "fh.write('fromNode\\ttoNode\\n')\n",
    "with open('union_diffmir_target.txt') as f:\n",
    "    for line in f:\n",
    "        line = line.rstrip()\n",
    "        lst = line.split('\\t')\n",
    "        \n",
    "        if lst[0] not in mirTarget.keys():\n",
    "            mirTarget[lst[0]] = []\n",
    "            \n",
    "        if lst[1].split('.')[0]+'.'+lst[1].split('.')[1] not in mirTarget[lst[0]]:\n",
    "            mirTarget[lst[0]].append(lst[1].split('.')[0]+'.'+lst[1].split('.')[1])\n",
    "        \n",
    "for ke, val in mirTarget.items():\n",
    "    for subv in val:\n",
    "        fh.write(ke+'\\t'+subv+'\\n')\n",
    "            \n",
    "            \n",
    "fh.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "76\n",
      "199\n"
     ]
    }
   ],
   "source": [
    "mirC = []\n",
    "targetC = []\n",
    "\n",
    "fh = open('cytoscape.all.node','wt')\n",
    "with open('cytoscape.all.edge') as f:\n",
    "    for line in f:\n",
    "        line = line.rstrip()\n",
    "        lst = line.split('\\t')\n",
    "        \n",
    "        if line.startswith('from'):\n",
    "            continue\n",
    "        \n",
    "        \n",
    "        if lst[0] not in mirC:\n",
    "            fh.write(lst[0]+'\\t'+'mir'+'\\n')\n",
    "            mirC.append(lst[0])\n",
    "            \n",
    "        if lst[1] not in targetC:\n",
    "            fh.write(lst[1]+'\\t'+'target'+'\\n')\n",
    "            targetC.append(lst[1])\n",
    "            \n",
    "            \n",
    "print (len(mirC))\n",
    "print (len(targetC))\n",
    "\n",
    "fh.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "anaconda-cloud": {},
  "kernelspec": {
   "display_name": "Python [Root]",
   "language": "python",
   "name": "Python [Root]"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.5.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}
